\documentclass[a4,center,fleqn]{NAR}

%\usepackage{NAR-natbib}
\usepackage{graphics}
\usepackage{subcaption}
\usepackage{verbatim}%for comments
\usepackage[table,x11names,dvipsnames]{xcolor}
\usepackage{adjustbox}% to resize figures
%\usepackage[skip=4pt]{caption}  % to keep dstance between caption and table/figure
%\usepackage[sort&compress]{natbib}
\usepackage{xspace}
%\usepackage{algorithm}
%\usepackage{algpseudocode}
\usepackage{amsmath}
\usepackage{xcolor}
\usepackage{todonotes}
\usepackage{booktabs}
\usepackage{relsize}
\usepackage{lipsum}
\usepackage{collcell}
\usepackage{pgfkeys}
\usepackage{amsmath}
\usepackage{varwidth}
\usepackage{tikz}
\usepackage{makecell}
\usepackage{gensymb}
\usepackage{inconsolata}



\usetikzlibrary{shapes.geometric, arrows, calc, fit, positioning}



\usetikzlibrary{shapes.geometric, arrows,calc}
%\tikzset { *|/.style={to path={	(perpendicular cs : horizontal line through ={(\tikztostart)},	vertical line through{(\tikztotarget)})	--(\tikztotarget) \tikztonodes	}	}	}

\pgfdeclarelayer{background}
\pgfdeclarelayer{foreground}
\pgfsetlayers{background,main,foreground}


\input{macros.tex}

%%%%%% Article Customization %%%%%%%%%%%%%%%%%%%%%%%%


\setlength{\parskip}{.3em}
% for multicolumn text centering 
\hfuzz=5pt


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


% Enter dates of publication
\copyrightyear{2019}
\pubdate{$\star$}
\pubyear{2019}
\jvolume{$\star$}
\jissue{$\star$}

%\articlesubtype{This is the article type (optional)}
\setcounter{secnumdepth}{3} % for subsection reference



\begin{document}

\title{\OurTool{}:  Integrative Probing Analysis of Nucleic Acids Empowered by Multiple Accessibility Profiles}
\author{%
Afaf Saaidi\,$^{1}$\footnote{Both first authors contributed equally to the study},
Delphine Allouche\,$^{2\,*}$,
Mireille Regnier\,$^{1}$,
Bruno Sargueil\,$^{2}$
and Yann Ponty\,$^1$% 
\footnote{To whom correspondence should be addressed.
Tel: + 33 1 77578095; Email: yann.ponty@lix.polytechnique.fr}}


\address{%
$^{1}$\,CNRS UMR 7161, LIX, Ecole Polytechnique, 1 rue Estienne d'Orves, 91120 Palaiseau, France
and
$^{2}$\,CitCoM, Cibles th\'erapeutiques et conception de M\'edicaments, CNRS, Universit\'e de Paris, 4 avenue de l'observatoire, 75006 PARIS, France}
% Affiliation must include:
% Department name, institution name, full road and district address,
% state, Zip or postal code, country

\history{%
Received July 1, 2019;
Revised $\star$;
Accepted $\star$}

\maketitle
\begin{abstract}
The manual production of reliable RNA structure models from chemical probing experiments benefits from the integration of information derived from multiple protocols and reagents. However, the interpretation of multiple probing profiles remains a complex task, hindering the quality and reproducibility of modeling efforts.

\noindent We introduce \OurTool{}, an automated method for the computational modeling of RNA structure from multiple probing reactivity profiles. Input profiles can result from experiments based on diverse protocols, reagents, or even a collection of mutants, and are jointly analyzed to predict the dominant conformations of an RNA.

\noindent \OurTool{} combines sampling, unsupervised learning, and multi-optimization, to produce secondary structure models that are both stable and well-supported by experimental evidences. The analysis of several reactivity, both publicly available and specifically produced experimentally, demonstrates the good performances of \OurTool, even in a mono probing setting. It confirms the potential of integrating multiple sources of probing data, leading to recommendations for the experimental design of informative probing assays. 

\noindent Availability: \OurTool{} is freely downloadable at

{\centering \url{\ipanemapurl}\\[.5em]}

\noindent \textbf{Contact:} \href{yann.ponty@lix.polytechnique.fr}{yann.ponty@lix.polytechnique.fr}

\noindent Supplementary information available at NAR online.
\end{abstract}

%%%%%%%%%%%%%% Introduction %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\section*{Introduction}


Historically used as a validation assays~\cite{Knapp1989}, enzymatic and chemical probing  is increasingly used in combination with computational methods to inform a rational prediction of secondary structure models for RNA~\cite{Mathews2004}. Such an integrated approach to structure modeling has led to sizable improvements in prediction accuracy~\cite{Washietl2012} and is currently at the core of successful modeling strategies~\cite{Miao2017}. However, the interpretation of probing data, to inform structure prediction, is challenged by a number of factors, including structural heterogeneity, experimental errors, structural dynamics and the potential variability of reactivity measurements across replicates. 

Reagents used within \Rev{selective 2 -hydroxyl acylation analyzed by primer extension (SHAPE)~\cite{Wilkinson2006} protocols} represent a popular class of small molecules. They react with the 2’-hydroxyl of flexible ribose~\cite{McGinnis2012}, although the exact properties observed by SHAPE remain the object of ongoing investigations~\cite{McGinnis2012,Sexton2017,Hurst2018,Mlynsky2018,Frezza2019,Busan2019}.
As ribose flexibility is proportional to the degree of freedom of the nucleotide, it is assumed that SHAPE reagents discriminate nucleotides in stable interactions from others. Different SHAPE reagents are endowed with different dynamics, that can help differentiate Watson-Crick base pairs from more dynamic tertiary contacts~\cite{Gherghe2008,Steen2012,Rice2014,Busan2019}. Other chemical probes, harbouring different chemical reactivities, have been developed. Amongst the most popular, DiMethyl Sulfate (DMS) is a small molecule that methylates N1 of adenines and N3 of cytosine if they are not involved in a hydrogen bond. Similarly, 1-cyclohexyl-3-(2-morpholinoethyl)carbodiimide metho-p-toluene sulfonate (CMCT) reacts on the Watson-Crick face of guanosine (N1) and uracyl (N3) when not engaged in a hydrogen bond~\cite{Ehresmann1987,Brunel2000}. These two reagents therefore mostly reveal Watson-Crick base pairing, but also any other type of contact involving the same edge. This diversity of probes not only increases coverage of the different positions and structural contexts in an RNA, but it also provides different qualitative information\Rev{, including in an \emph{in vivo} setting~\cite{Wang2019}}. It has been widely used since the very early days of RNA structure probing~\cite{Moazed1986, Ehresmann1987, Romaniuk1988, Butcher1994, Brunel2000,  Cordero2015, Somarowthu2015, Gross2017}. 

% **************************************************************
% Keep this command to avoid text of first page running into the
% first page footnotes
\enlargethispage{-50.1pt}
% **************************************************************


Computationally, the reactivity of a nucleotide is typically used as a proxy \Rev{to assess the unpaired nature of individual nucleotides}. The last couple of decades have nevertheless seen a series of paradigm shifts in the ways probing information is integrated, somehow mirroring the evolution of {\em ab initio} methods for secondary structure prediction. The seminal work of Mathews~\cite{Mathews2004} used cutoffs to transform reactivity values into hard constraints. Depending on the reagent/enzyme, significantly unreactive or reactive positions were forced to either paired or remain unpaired respectively within predicted structures. However, such \emph{hard} constraints turned out to be very sensitive to the choice of the cutoff, easily leading to artificially unpaired predictions or unsatisfiable sets of constraints. With the advent of SHAPE probing, subsequent methods lifted the requirement of thresholds and integrated reactivities directly as pseudo-energies, supplementing the Turner nearest-neighbor energy model~\cite{Turner2010} within an (pseudo) energy minimization scheme~\cite{Deigan2009,Zarringhalam2012,Washietl2012}. Such methods can be seen as optimizing a tradeoff between the thermodynamic stability of a model, assessed by the Turner energy model, and its compatibility with the reactivity profile. As a consequence, popular packages for secondary structure prediction, such as \Software{RNAStructure}~\cite{Mathews2004} or the \Software{Vienna} package~\cite{Lorenz2011}, now routinely accept SHAPE-like reactivity profiles as input of their main structure prediction methods. \Rev{Besides free-energy optimization, such methods notably} include the joint folding of aligned RNAs~\cite{Lavender2015}, partition function, statistical sampling and Maximal Expected Accuracy predictions~\cite{Spasic2017}\ldots 


However, independent computational predictions for the same RNA using probing data obtained with different probes often yield substantially different models, none of which are fully consistent with all the probing data. 
It follows that, while theoretically informative, a multiprobing strategy often leaves a user with different models, from which one cannot objectively decide. In order to identify the best fit with all the data, researchers resort to very intuitive and manual methods such as projecting the results obtained with one probe onto the different models obtained using the constraints from another probe~\cite{Herbreteau2005,James2008,Weill2004}. At the end of the modeling process, the modeler is often left with several alternatives, all of which may appear equally consistent with the probing data. Thus, we have developed a new modeling procedure that jointly takes into account multiple probing data, and ultimately yields a small collection of secondary structure models.


%As a consequence, integrating the probing data as "hard" constraints, prohibiting any pairing for highly reactive nucleotides or forcing pairing of the unreactive, would preclude the prediction of valuable structure. For this reason it appeared more relevant and efficient to take probing data into account in the prediction pipeline as soft constraints, i.e as penalties in the score when elaborating the models~\cite{Deigan2009,Washietl2012,Zarringhalam2012}. 
% **************************************************************
% Keep this command to avoid text of first page running into the
% first page footnotes
%\enlargethispage{-65.1pt}
% **************************************************************



% **************************************************************
% Keep this command to avoid text of first page running into the
% first page footnotes
%\enlargethispage{-65.1pt}
% **************************************************************





%{\noindent\bf Contribution je ne mettrais pas de "contribution", c'est une intro pas un résumé ni une conclusion.}


\begin{comment}
\begin{itemize}
\item Complementarity of different probing data (protocols, reagents\ldots)
\item Auto-correlation of probing signal is important (partial validation of this claim due to partial data)
\item Why MEA can be bad... and the sensitivity of SHAPE experiments (dot-plot+ detection of outliers, PCA $\to$ Case study). Sampling from SHAPE modified ensembles induces higher robustness with respect to ``outlier'' (rotten) experiments
\item (wishful thinking) Despite relying on a stochastic sampling, results are reproducible
\item (wishful thinking) Performances of our methods are impacted by the choice of the chosen clustering/$\#$clusters
\item (wishful thinking) Increase of performances upon repeated experiments
\item (wishful thinking) Recommendations for experimentalists  (choosing the right combination of probes)
\end{itemize}
\end{comment}


\section*{Material and methods}

\subsection*{The \OurTool{} method}
We introduce the Integrative Probing Analysis of Nucleic Acids Empowered by Multiple Accessibility Profiles (\OurTool{}), a novel approach that integrates the signals produced by various chemical probes \Rev{to predict one or several secondary structure models for a given RNA.}

It performs a structural clustering across multiple sets of structures sampled in different experimental conditions, and ultimately returns a set of structures that are representative of the dominant conformations supported across conditions. Its underlying rationale is that the prominent presence of a stable secondary structure within the  (pseudo-)Boltzmann ensembles induced by multiple experimental conditions should increase its likelihood to be (one of) the native structure(s) for a given RNA. It is thus hoped that integrating several reactivity profiles may be used to promote the native structure as one of the dominant structures within the multi-ensemble, and help circumvent the limitation of pseudo-energies derived from single reactivity profile, which are generally not sufficient to elect the native structure as its minimum (pseudo)-free energy candidate. In other words, combining ensembles of structures generated using multiple probing experiments is likely to denoise the Boltzmann (multi-)ensemble, and thus mitigate systematic biases induced by experimental conditions and reagents.

\begin{figure}
	{\centering\resizebox{.9\columnwidth}{!}{
			\input{diagram_NAR.tex}
		}\\}
	
	\caption{\OurTool{} workflow: \OurTool{} takes as input an RNA sequence with profiling data, denoted by reactivities, from various experimental conditions. \OurTool{} proceeds, first,  with a stochastic sampling that results into samples of predicted secondary structure. The data-driven predicted structures are then gathered in one sample, serving as input for the clustering step. \OurTool{} proceeds, then, with an iterative clustering that ends once \Rev{a stopping criterion is} reached. This step allows to identify the adequate number of clusters $k$ to be considered. The $k$ resulting clusters are then \Rev{represented} by their centroid structures. Clusters figuring on the 2D-Pareto frontier are considered to be optimal and subsequently their corresponding centroids are reported as the predicted structure through \OurTool.}\label{fig:approach}
\end{figure}


\paragraph{Sampling the pseudo-Boltzmann ensembles.} Our method, summarized in Figure~\ref{fig:approach}, takes as input a set $\mathcal{D}$ of probing experiments, each materialized by a reactivity profile. It starts by producing (multi)sets of representative structures for each of the reactivity profiles using a SHAPE-directed variant of the classic Ding-Lawrence algorithm~\citep{Ding2003}. Following Deigan\etal\cite{Deigan2009}, \Def{soft constraints} are used to complement the free energy contributions of the classic Turner energy model with pseudo energy contributions resulting from the reactivity derived from a probing experiment.  Given a reactivity $r_i$ for a position $i$ in a probing $d$, we associate a free-energy bonus to unpaired positions, defined as
$$\Delta G_d(i) = m \log(r_i +1 )+b$$ 
using $m=1.3$ and  $b=-0.4$. \Rev{Those values were halved in comparison to those recommended by Deigan\etal\cite{Deigan2009}, following a grid search optimization on the Cordero dataset, and based on the rationale that lower absolute values for pseudo-energy bonuses increase the expected overlap between pseudo-Boltzmann ensembles.} Those pseudo energy contributions effectively guide our predictions towards a subset of structures that are in good agreement with probing data. For each condition $d$ in $\mathcal{D}$, we use the soft constraints framework~\cite{Lorenz2016} of the \Software{RNAfold} software~\cite{Lorenz2011} to produce a random (multi)set $\mathcal{S}_d$ of $\NumStruct$ structures in the pseudo-Boltzmann ensemble.




\Rev{\paragraph{Clustering across conditions.} In order to infer recurrent conformations across sampled sets, \OurTool{} agglomerate structure (multi)sets while keeping track of their condition of origin, and \Def{clusters} with respect to the \Def{base-pair distance}, the number of base pairs differing between two structures. A clustering algorithm then partitions the (multi)set of sampled structures into \emph{clusters}, (multi)sets of structures such that the accumulated sum of distances over clusters is minimized.

Among the many available options, we chose the Mini Batch $\NumClust$-Means algorithm~\citep{Sculley2010} (MBkM), implemented in the \Software{scikit-learn} pacakage~\cite{Pedregosa2012}, which requires less computational resources than the classic $\NumClust$-means algorithm, yet performed similarly in preliminary studies as an extensive collection of both agglomerative (affinity propagation) and hierarchical (Ward, Diana, McQuitty) clustering algorithms. A dissimilarity matrix, presenting the pairwise base-pair distance between structures, is precomputed and fed to the clustering algorithm.

Any cluster $C$ output by the clustering is a multiset of structures, each labeled with its origin condition of $\mathcal{D}$.
The \Def{cluster probability of a structure feature} $f$ (base pair or unpaired base) within a cluster $C$ is then defined as 
$$ \mathbb{P}_C(f) = \frac{\sum_{\substack{S\in \mathcal{C}\\\text{s.t. } f\in S}} e^{-E(S)/RT}}{\sum_{\substack{S'\in \mathcal{C}}} e^{-E(S')/RT}} $$ 
where $R$ represents the Boltzmann constant, $E(S)$ is the Turner free-energy, and $\mathcal{C}$ is the (non-redundant) set of structures in $C$. From those probabilities we define the \Def{centroid structure} of a cluster as its Maximum Expected Accuracy (MEA) structure, computed efficiently following Lu\etal~\citep{Lu2009}.

Moreover, define the (pseudo-)\Def{Boltzmann condition probability} of a structure $S$, generated for a probing condition $d$ as part of a sampled  set $\mathcal{S}_d$, as
$$\mathbb{P}_d(S) = \frac{e^{-E_d(S)/RT}}{\mathcal{Z}^*_d} \text{, with } \mathcal{Z}^*_d := \sum_{S'\in \mathcal{S}_d} e^{-E_d(S')/RT}$$
where $E_d(S)$ is the pseudo free-energy, including the Turner free-energy, assigned to the structure $S$ within the probing condition $d$.
The \Def{stability} of a cluster $C$ denoted its accumulated pseudo-Boltzmann probability across conditions, computed as
$$\text{Stability}(C) = \sum_{d\in \mathcal{D}} \sum_{S\in C\cap \mathcal{S}_d}  \mathbb{P}_d(S).$$
A cluster is deemed \Def{significantly populated} if its stability exceeds a predefined threshold $\epsilon$. 
We set  $\epsilon =\Rev{|\mathcal{D}|}/3$ by default, such that at most three clusters are deemed significantly populated, and used as our primary candidates.
Finally, we consider two clusters to be \Def{highly similar} if their centroid structures differ by at most $\delta$ base pairs ($\delta=1$ by default), allowing the identification of clusters in the presence of minor variations.




The targeted number of clusters is a critical parameter of the \CL{} algorithm. It should, at the same time, remain small enough to ensure reproducibility, while being sufficiently large to discriminate outliers and ensure consistency within each cluster. We determine an \Def{optimal number of clusters} $\NumClust$ using an iterative heuristics, gradually increasing the number of clusters until a significantly-populated cluster is split into two similar clusters, or a poorly populated (outlier) cluster is created.
Namely, our iterative heuristic consists in running \CL over an increasing number $\NumClust$ of clusters, starting with $\NumClust=2$, until the following \Def{stopping criterion} is met: 1) Two significantly populated clusters have associated centroids which are highly similar; or 2) Centroid structures of significantly populated clusters from the previous iteration are highly similar to those of the current iteration.}


\paragraph{Filtering the promising conformer(s).} Finally, we \Def{elect the most promising cluster(s)}, and return their centroid(s). While the final number of clusters may potentially be large, only a handful of clusters are expected to represent structures that \Rev{are} both stable, and supported by a large number of conditions. The remaining clusters are indeed probably artifacts of the clustering method, but nevertheless useful to filter out \emph{noisy} structures. 

\Rev{We postulate that a perfect cluster should have large stability, as defined above, and be representative of several conditions.} In the presence of a set of experimental conditions $\mathcal{D}$, we consider that a cluster $C$ supports a given condition $d$ when its probability within $d$ exceeds a given threshold $\tau$. The number of conditions supporting a cluster $C$ is defined as
$$\text{Support}(C) = \left|\left\{d\in \mathcal{D} \mid \left(\textstyle\sum_{S\in C\cap \mathcal{S}_d} \mathbb{P}_d(S)\right)\ge \tau \right\}\right|.$$
The value of $\tau$ is set to $1/(\NumClust+1)$ \Rev{with $\NumClust$ the final number of clusters},  ensuring at least one supporting cluster for each condition.

\OurTool evaluates the Stability and Support metrics for each cluster, and filters out any cluster that is dominated by some other with respect to both metrics. The remaining ones are \Def{Pareto optimal}, a classic concept in multi-objective optimization~\cite{Mattson2005}. \OurTool computes and returns the MEA centroid~\cite{Lu2009} of the Pareto-optimal clusters as its final prediction(s).



\subsection*{Pairwise comparison of structural ensembles induced by reactivity profiles}\label{sec:dotplots}

We want to compare the structural ensembles induced by reactivity profiles, produced across diverse experimental conditions. To that purpose, we simply consider the base pair probability matrices, or dot-plots, resulting from supplementing the Turner energy model with pseudo energy terms. %This can be seen as a projection of experimentally constrained conformational spaces. 
Dot plots can be computed efficiently in the presence of pseudo-energy terms using a variant of the McCaskill algorithm~\citep{McCaskill1990}.

%To find the most compatible structural models across miscellaneous probing data, 
%As a first assessment of the structural compatibility across miscellaneous probing data, we compared their respective Boltzmann probability distributions where the pseudo-Boltzmann probability to observe a given structure from the ensemble is deduced from the base-pair probabilities computed through a recursive algorithm ~\cite{McCaskill1990}. 
%We remind that RNA folding process is a stochastic mechanism. For a given RNA a certain number of possible stable conformations is present in the ensemble.
%Assuming a Boltzmann thermodynamic equilibrium and the stability of  conditions, a Boltzmann probability to observe each secondary structure from the ensemble could be calculated~\cite{McCaskill1990}. 

As a measure of the \Def{ensemble distance} $\Edist$ induced by probing data, we consider the dot-plots associated with experimental conditions $d$ and $d'$, and compute the squared Euclidean distance, such that
$$\Edist (d,d')=\sum\limits_{i=1}^\RL\sum\limits_{j=i+1}^\RL \left(\mathbb{P}(i,j\mid d)-\mathbb{P}(i,j\mid d')\right)^2.$$
with $\RL$ the length of the RNA sequence,  $\mathbb{P}(i,j\mid d)$ the Boltzmann probability of forming a base pair $(i,j)$ in the pseudo-Boltzmann ensemble associated with condition $d$.


Individual dot-plots were computed using the \Software{RNAfold} software in the \Software{Vienna Package 2.2.5}, using the \Software{-p} option in combination with the pseudo-energy terms introduced by Deigan\etal\cite{Deigan2009}. 

% Yann:
%\begin{itemize}
% \item Sampling representative sets of structures. Pseudo potentials et distribution de Boltzmann (Utiliser refs, justifier et positionner)
%\item Clustering and extraction of representative structures (inc. optimal \#clusters)
%\item Selecting a subset of dominant clusters
%\item Comparing pseudo-Boltzmann ensembles
%+illust + On peut faire ça uniquement parceque les ARN sont de meme taille, ie l'alignement est sans ambiguite.
%Citer Halvorsen et al (+Laederach Plos CB 2013?)%
%\item Generating artificial SHAPE datasets
%Citer Suskozd (on generalise et etend au dela des grands rRNAs) et simulateur SHAPE (ref???)
%\end{itemize}


% \subsection*{Generative model for probing data}

%Many models to simulate SHAPE probing data starting from the analysis of the SHAPE reactivity distribution performed by ~\cite{Sukosd2013}, have been suggested.
%ull Access% c'est mal dit!===
%In order to generalize the simulation of probing data to other probes than SHAPE and to suggest more deterministic models, we build the normalized distribution of reactivities in function of three structural categories: {\em Helix, Helix-End} and {\em unpaired} using reactivity profiles from 6 RNAs with known structures and three available probing data ~\cite{Cordero2012}.  
%We constructed probabilistic models to simulate probing data from the different distributions; a number of 500 bins was set, the mid-value for each bin with the corresponding probability was considered as parameter for the generative model. Consequently, a random generator was implemented to simulate probing data for each single structural category and each specific probing method where the generator input is the mid-value vector w probability vector.


\subsection*{Datasets} 
\label{sec:datasets}
To validate our computational method quantitatively, we consider several datasets, depending on the availability of probing data for one or several reagents, restricted to the wild type or produced for several point-wise mutants. Each dataset consists of sequences and individual reactivities to one or several probes, at each position in the RNA, completed with one or several functionally-relevant secondary structures.

\paragraph{Hajdin dataset.} 
A dataset was gathered by Hajdin\etal\cite{Hajdin2013} to validate the predictive capacities of probing data-driven predictions. It consists of $24$ RNA sequences with known secondary structures for which a single chemical probing reactivity profile (1M7-\SH) is available.
This dataset includes sequences originating from a variety of organisms, and spans lengths ranging from $34$ nts to $530$ nts, with a focus on riboswitches and complex RNA architectures. 


\paragraph{Cordero dataset.} 
In order to test the integration of different probing sources, we consider a dataset introduced in Cordero\etal\cite{Cordero2012}, consisting of $6$ RNAs with known structures, for which reactivities are available for three reagents: {\tt NMIA-\SH, DMS} and {\tt CMCT}. 
Probing data were downloaded from the RMDB~\citep{Cordero2012a} on July 2017. In the RMDB, reactivity scores are reported for all nucleotides, including those that are not expected to react with a given reagent. Thus, for the DMS (resp. CMCT) probing, we restricted reactivities to positions featuring nucleotides {\sf A} and {\sf C} (resp. {\sf G} and {\sf U}), setting the reactivity to 0 for other positions. This assumption allowed to decrease the noise generated by reactivities associated with non-targeted nucleotides, leading to more accurate predictions (data not shown). 



\paragraph{Didymium structural model and probing data (\didy{} dataset).} 
We considered the 188 nucleotides Lariat capping ribozyme from {\itshape Didymium iridis}, resolved 3.85 \AA{} resolution using X-ray cristallography (PDB: 4P8Z)~\citep{Meyer2014}.  We annotated the secondary structure elements using the \Software{DSSR} software from the 3DNA suite~\cite{Lu2015}. \Rev{Non-canonical base pairs were removed, and a non-pseudoknotted secondary structure was extracted as the maximum subset of non-crossing base pairs~\citep{Smit2008}.}

Probing data were experimentally generated, as described in the next section, using a comprehensive set of conditions covering some of the popular probing reagents and SHAPE technologies. 
We also considered the presence/absence of Mg$^{2+}$, both to assess the capacity of \OurTool{} to recover tertiary interactions, and to assess the induced discrepancy on probing profiles and pseudo-Boltzmann ensembles.


\paragraph{Cheng dataset.} 
Starting from the assumption that a functional structure should be preserved during evolution, we wanted to assess the agreement that might exist between probing data profiles for a set of RNA mutants. 
We considered DMS probing data, generated by~\cite{Cheng2017} through systematic point-wise mutations, for the Lariat-capping ribozyme (equivalent to \Cond{DMS}{mg}{MaP}{ilu}{} in the nomenclature below). We renormalized each reactivity profile following the method introduced by Deigan\etal\cite{Deigan2009}, restricted to the primer-free sequence: values greater than $1.5 $ times the interquartile range were discarded, and remaining values were divided by the mean of the top $10\%$ reactivities.  Overall, this constitutes a collection of 188 sequences, each having its associated reactivity profile.



\subsection*{Experimental probing protocols}

\Rev{To systematically assess the potential of multiple sources of probing, we considered a difficult example, the \textit{Didymium iridis} Lariat Capping ribozyme (\didy{}). The native structure of \didy{}, shown in Figure~\ref{fig:didy} is highly complex, and features two pseudoknots which cannot be explicitly modeled by most computational methods, making \didy{} a challenging target for secondary structure prediction.}


\didy{} was probed with different SHAPE reagents: 1M7 (1-methyl-7 nitrosatoic anhydre), NMIA (N-Methyl Isatoic Anhydre), BzCN (Benzoyl cyanide) and NAI (2-methylnicotinic acid imidazolide) in presence and absence of Mg$^\text{2+}$. \Rev{\didy{} was also probed with DMS (DiMethylSulfate) and CMCT, this time only in presence of Mg$^\text{2+}$, resulting in a total of 16 \Def{probing conditions}, a term we use in the following to denote a combination of probing technology, reagent and presence/absence of Mg$^\text{2+}$ (+ sequencing technology). For each probing condition, three experiments were performed in presence/absence of the reagent, and in a denatured context, following classic SHAPE protocols~\cite{Wilkinson2006,Smola2015}.}

As a preliminary experiment, we verified that \didy{} generally adopts a single global architecture. To this end, we subjected \didy{} to a standard denaturation protocol (80\degree{}C for 2 min in H$_\text{2}$O, addition of 40 mM of HEPES at 7.5 pH, 100 mM of KCl, 5 mM of MgCl$_\text{2}$, followed by 10 min at room temperature, and 10 min at 37\degree{}C), and observed the production of a single band on a non-denaturing PAGE, strongly suggesting the adoption of a single conformation.

\paragraph{Stops-inducing probing protocol (SHAPE-CE).} 
6 pmol of RNA were resuspended in 18 $\mu$l of water, denatured at 80\degree{}C for 2 minutes and cooled down at room temperature during 10 minutes in the probing buffer (40 mM HEPES at 7.5 pH, 100 mM KCl, in presence or absence of 5 mM MgCl$_\text{2}$. After a 10 minutes incubation at 37\degree{}C, RNAs were treated with 2 mM of SHAPE reagent or DMSO (negative control) and incubated for 2 (BzCN), 5 (1M7), 30 (NMIA) or 60 (NAI) minutes at 37\degree{}C. Modified or unmodified RNAs were purified by ethanol precipitation and pellets were resuspended in 10 $\mu$l of water.
Modifications were revealed by reverse transcription using 5' fluorescently labelled primers (D2 or D4 WellRED, Sigma Aldrich 5' CTG-TGA-ACT-AAT-GCT-GTC-CTT-TAA 3') and M-MLV RNAse (H-) reverse transcriptase (Promega) as previously described~\cite{Deforges2017}, with only minor modification of the originally described SHAPE protocol~\cite{Wilkinson2006}. cDNAs were separated by capillary electrophoresis (Beckman Coulter, Ceq8000). Data were analyzed using the  \Software{QuSHAPE}~\cite{Karabiber2013} software. RNA probing was performed in triplicate with distinct RNA preparations. The corresponding data sets are named after the probe followed by {\sc CE}, and {\sc mg} if probed in presence of MgCl$_\text{2}$.

\paragraph{Mutations-inducing probing (SHAPE-MaP).}
Probing was conducted as described in Smola\etal\cite{Smola2015}, except we used SuperScript III reverse transcriptase (ThermoFisher) at 50\degree{}C for 3 hours using the following specific primer: 

{\centering \relsize{-1}%
5' CTT-CAT-AGC-CTT-ATG-CAG-TTG-CTT-TTT-TTT-TTT
   TTT-TTT-TTT-GAT-TGT-CTT-GGG-ATA-CCG-GAT 3'\\}

For \OneMSevILUThreeMg, the denaturation step was repeated three times : After a 1 minute incubation at 95\degree{}C, 10 mM of 1M7 was added and the mix was incubated 1 minute at 95\degree{}C. This step was repeated 3 times for the samples labeled 3D. For \NMIA and \NMIAMg, experiments were conducted as described above, except that the NGS library preparation was adapted to Ion Torrent sequencing. Sequences were mapped and analyzed with \Software{ShapeMapper2}~\cite{Busan2018} for \OneMSevILUThreeMg and \OneMSevILUThreeMg. For the other conditions they are mapped with a script based on \Software{ShapeMapper}~\cite{Smola2015}, but adapted to Ion Torrent output files. The corresponding data sets are named after the probe followed by {\sc MaP}. The presence of {\sc mg} as a superscript indicates the presence of MgCl$_\text{2}$, while the subscript {\sc il} (Illumina) or {\sc it} (Ion Torrent) specifies the sequencing technology. 


\paragraph{DiMethylSulfate (DMS) and CMCT probing.}
DMS and CMCT probing were conducted essentially as described in~\cite{Weill2004,James2008}. Succinctly, 6 pmol of RNA were resuspended in 18 $\mu$l of water, denatured at 80\degree{}C for 2 minutes and cooled down at room temperature for 10 minutes in a probing buffer for DMS probing (40 mM HEPES pH 7.5, 100 mM KCl, 5 mM MgCl$_\text{2}$) or in CMCT (50 mM of potassium borate pH 8, 100 mM KCl and 5 mM MgCl$_\text{2}$). For DMS, RNA was then treated with 1 $\mu$l of DMS (1:12 in ethanol) or 1 $\mu$l of ethanol for 5 minutes at 37\degree{}C (mock reaction), the reaction was stopped by addition of 400 mM of Tris at 7.5 pH and immediately put on ice. For CMCT,  RNA was treated with 10 $\mu$l of CMCT (42 mg/mL) or 10 $\mu$l H$_\text{2}$O for 10 minutes at 37\degree{}C (mock reaction). Modified RNA were ethanol-precipitated and resuspended in 10 $\mu$l of water. Modifications were mapped as described above for SHAPE-CE experiments.The corresponding data sets are named after the probe, followed by {CE} and {\sc mg}.


\subsection*{Benchmarking methodology}

The Matthews Correlation Coefficient (MCC) is a classic metric for assessing the quality of a predicted structure $S$, identified by a set of base pairing positions, based on its agreement with an accepted reference (native) structure $\Ref$. It can be interpreted as a compromise \Rev{between the main metrics derived from the confusion matrix}, and is defined as 
$$\text{MCC}(S\mid \Ref)=\frac{\text{TP} \times \text{TN} -\text{FP} \times \text{FN}}{\sqrt[]{(\text{TP}+\text{FP})(\text{TP}+\text{FN})(\text{TN}+\text{FP})(\text{TN}+\text{FN})}},$$
where TP, FP, TN, FN classically represent the correctly/erroneously predicted and correctly/erroneously omitted base pairs in $S$ with respect to $\Ref$.

For the sake of direct comparison with some competing methods~\cite{Spasic2017}, we also report the Geometric Mean (GM) metric, defined as:
$$ \GMean(S\mid \Ref)=\sqrt{\text{Sens}(S\mid \Ref)\times \text{PPV}(S\mid\Ref)} $$
where $\text{Sens}(S\mid \Ref)$ represents the proportion of base pairs in $\Ref$ that are in $S$, and $\text{PPV}(S\mid \Ref)$ is the proportion of base pairs in $S$ that are also in $\Ref$. This quality metric has very high correlation with the MCC, to which it is equivalent for large values of $n$, as shown by~Gorodkin\etal\cite{Gorodkin2001}.

\Rev{
\subsection*{Assessment of statistical significance}
Following Xu\etal\cite{Xu2012}, we assessed the statistical significance of observed changes in predictive performances, using a two-tailed Student $t$-test with unequal variance, using a classic significance threshold $\alpha=5\%$. When comparing performances within a given dataset, we used the paired version of the test.


Intuitively, the $t$-test estimates the probability that an observed change in performance can be attributed to chance alone.
It considers the hypothesis $H_0$ that the two distributions have equal mean and computes a \Def{P-value}, \emph{i.e.} the likelihood of the observed values under $H_0$. If the P-value is lower than $\alpha$, then $H_0$ is deemed \Def{refuted}, and the observed difference in average performance is considered as significant.

}

\section*{Results}

\subsection*{Implementation}

\OurTool{} was implemented in \Software{Python 2.7+}, and mainly depends on the \Software{RNAsubopt} software in the Vienna package~\cite{Lorenz2011}, and \Software{scikit-learn}~\cite{Pedregosa2012}. \OurTool{} is free software distributed under the terms of the MIT license, and can be freely downloaded, along with a collection of helper utilities, from:

{\centering \url{\ipanemapurl}\\}

\subsection*{Considering multiple probing conditions \Rev{appears to} improve the quality of predictions}\label{sec:cordero}

%In order to validate our assumption of an enrichment in the presence of multiple probing profiles, we analyzed the Cordero \emph al} dataset, which contains CMCT, DMS and NMIA probing data for 6 RNAs. 
%


\colorlet{BadCol}{Burlywood1!70!red}

\begin{table*}
	\begin{adjustbox}{max width=\linewidth}%
		\setlength{\fboxsep}{0pt}
		%		\newcommand{\Base}[1]{\colorbox{blue!30}{\strut#1}}
		\newcommand{\Base}[1]{\colorbox{Aquamarine3!60}{\strut#1}}%
		\newcommand{\G}[1]{\colorbox{Aquamarine3!60}{\strut#1}}%
		\newcommand{\B}[1]{\colorbox{BadCol}{\strut#1}}%
		\begin{tabular}{@{}llr@{}} \toprule
			Sequence& \tt {GAUAUGAGGAGAGAUUUCAUUUUAAUGAAACACCGAAGAAGUAAAUCUUUCAGGUAAAAAGGACUCAUAUUGGACGAACCUCUGGAGAGCUUAUCUAAGAGAUAACACCGAAGGAGCAAAGCUAAUUUUAGCCUAAACUCUCAGGUAAAAGGACGGAG}
			& GM\\
			\midrule
			$\varnothing$& {\tt \G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{.}\B{(}\B{(}\B{(}\B{(}\B{(}\G{.}\B{.}\B{.}\B{(}\B{.}\B{.}\B{.}\G{.}\B{.}\B{.}\B{)}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\B{.}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\B{.}\B{.}\B{(}\G{(}\G{(}\B{(}\G{.}\G{.}\G{.}\B{)}\G{)}\G{)}\B{.}\B{.}\B{.}\G{.}\B{.}\G{(}\G{(}\G{.}\G{.}\G{.}\B{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\B{)}\G{.}\G{.}\G{)}\G{)}\B{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}}&.568\\
			CMCT & {\tt \G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{(}\G{(}\G{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\G{.}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\B{(}\G{.}\G{.}\B{(}\G{.}\G{.}\B{(}\B{(}\B{(}\B{(}\B{.}\B{.}\B{.}\B{(}\B{(}\B{(}\B{(}\B{(}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\B{(}\G{.}\G{.}\G{.}\B{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\B{.}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\B{.}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\G{.}\G{.}\G{.}\B{)}\G{.}\G{.}\B{)}\B{.}\B{.}\B{.}\B{.}}&.627\\
			DMS+CMCT & {\tt \G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{(}\G{(}\G{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\G{.}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\B{(}\B{(}\B{.}\B{.}\B{.}\B{.}\B{(}\B{(}\B{(}\B{(}\B{(}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\B{(}\G{.}\G{.}\G{.}\B{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\B{.}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\B{.}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\B{)}\B{)}\B{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\B{.}\B{.}\B{.}\B{.}\B{.}}&.658\\
			DMS+CMCT+NMIA & {\tt \G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{(}\G{(}\G{(}\B{(}\B{(}\B{(}\B{(}\B{(}\B{.}\B{.}\G{.}\G{.}\G{.}\G{.}\B{)}\B{)}\B{)}\B{)}\B{)}\G{.}\G{.}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{(}\B{(}\G{.}\G{.}\G{.}\B{)}\G{)}\G{)}\G{)}\G{)}\G{)}\G{.}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\B{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{(}\G{(}\G{(}\G{(}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\B{)}\G{.}\G{.}\G{)}\G{)}\G{)}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{.}\G{)}\G{)}\G{)}\G{)}\G{)}} & \textbf{.868}\\ \midrule
			Reference& {\tt \Base{((((((((......((((((....)))))).(((....(((.....)))...)))........))))))))........(((((......(((((.....))))).(((....(((....((((....)))).....)))...))).......)))))}}& 1\\
			\bottomrule
		\end{tabular}
	\end{adjustbox}\\
	\caption{Dot-bracket representations of the native structure and \OurTool{} predictions for the glycine riboswitch sequence of the Cordero\etal\cite{Cordero2012} data set, supplemented with an increasing collection of probing conditions. Positions with green and red backgrounds indicate correctly and incorrectly predicted base pairs.}\label{fig:glycine_example}
\end{table*}


Since \OurTool{} supports any number of probing profiles, we considered the Cordero\etal\cite{Cordero2012a} dataset, further described in the previous section. It consists of 6 RNAs of known structures, for which reactivity profiles are available for the CMCT, NMIA and DMS reagents. We executed \OurTool{} with default parameters on each sequence and any subset of the three available conditions. The centroid secondary structure  associated with the largest probability cluster was considered as the final prediction. Predictions were evaluated in term of the geometric mean (GM) metric, for the sake of comparison with previous studies~\cite{Spasic2017}. As a control experiment, we also report ${\tt RNAfold}$ predictions in presence/absence of probing data, both in energy minimization (MFE) and maximum expected accuracy (MEA) modes. Figure~\ref{cordero1} shows the averaged GM values for all combinations of tools and probing data (Detailed results can be found in Supp. Tables S1 and S2).



%* Ajouter des conditions de probing -> meilleurs conditions
\begin{figure}
	{\centering{\includegraphics[trim=.5cm .3cm .2cm .2cm, clip, width=\linewidth]{graphs/cordero_mean}}\\}
	
	\caption{\Rev{Predictive performances (GM) of \OurTool{} on the Cordero~\cite{Cordero2012a} dataset consisting of reactivities associated with the CMCT, NMIA and DMS probing for 6 reference RNAs. (Pseudo) MFE and MEA structures, predicted using \Software{RNAfold}~\cite{Lorenz2011}, are included as references in the case of a single condition, and in the absence of probing.}}
	\label{cordero1}
\end{figure}


In the absence of probing data, MFE predictions are generally dominant on this dataset, trailed by the MEA, and followed by \OurTool which, in this setting, devolves into the classic Ding-Lawrence algorithm~\citep{Ding2003}. However, whenever probing reactivities are available, the single centroid returned by \OurTool always achieves higher GM values (Avg: $70\%$) than both MEA (Avg: $62\%$) and MFE  (Avg: $58\%$), whose relative performances depend on the probing reagent. %la moyenne sur les 6 ARNS et sur les 3 reactifs
Interestingly, the quality of MFE and MEA predictions does not systematically benefit from additional probing data. Indeed, for half of the reagents and methods, the average GM obtained in the presence of a single reagent is lower than in the absence of probing data. Also, the impact of single probing data varies greatly across the three reagents, and the GE values of predictions respectively informed by CMCT, DMS and NMIA are ordered increasingly for all approaches, except for a minor reversal of DMS and NMIA in MEA mode.


The joint analysis of pairs of probing conditions appears to average the quality of predictions. Any combination of probing data yields a GM value that is always greater than the worst-performing condition in the pair, yet worse than the best-performing alone. Interestingly, the addition of the worst performing condition (CMCT), does not equally affect the performances of DMS ($76\% \to 70.5\%$) %DMS \to DMS+CMCT
and NMIA  ($77.1\% \to 76.9\%$), %NMIA \to NMIA+CMCT
despite the latter conditions inducing similar GM values when considered alone. Indeed, CMCT+DMS yields GM values that are only remotely better than the worst-performing CMCT alone  ($70\% \to 70.5\%$), while CMCT+NMIA greatly outperforms CMCT alone  ($70\% \to 76.9\%$), almost matching the performance of the NMIA alone ($77.1\%$). It is also worth mentioning that NMIA+CMCT, combining the best and worst conditions, achieves a better combined performance than DMS+NMIA, the two best mono-probing conditions. 


Remarkably, the combination of the three conditions leads to the best overall predictions, averaging $78.5\%$ GM.
This \Rev{seemingly} improves by $8.5\%$, $2.7\%$ and $1.4\%$ over the average predictions achieved using CMCT, DMS and NMIA respectively.
%CMCT/DMS/NMIA :0.6988333333/0.7588333333/0.771
Table~\ref{fig:glycine_example} illustrates the incremental refinement of \OurTool predictions for a glycine riboswitch upon increasing the number of considered probing conditions.
\Rev{However, these observations merely suggest an improvement, as statistical significance cannot be established on such a small dataset.}
Still, they suggests some level of complementarity between probing conditions, as already suggested by recent analyses~\citep{Brunel2000,Steen2012,Rice2014,Yu2018}  and supported by further analyses in this paper.

 




%\clearpage



\subsection*{\OurTool outperforms state-of-the-art predictive methods when informed by a single profile}
%Mono-probing Mathews (Hadjim)
%* On est meilleur sur la version courante de RSample
%* Multi conformation
We assessed the predictive capacities of \OurTool in a classic setting where a single probing condition is available, and compared its performances against \Software{RSample} software recently introduced by Spacic\etal\cite{Spasic2017}, \Software{RSample} relies on a sampling/clustering method developed independently from the current work. It was shown to perform \Rev{comparably as} a comprehensive collection of state-of-the-art methods in probing-guided structure prediction, including \Software{RME}~\citep{Wu2015}, \Software{RNAprob}~\citep{Deng2016}, \Software{RNAprobing}~\citep{Washietl2012}, \Software{RNAsc}~\citep{Zarringhalam2012} and the \Software{fold} utility from the \Software{RNAStructure} suite~\citep{Reuter2010}.
We considered the Hajdin dataset~\cite{Hajdin2013}, which consists of 24 RNAs of lengths ranging from 47 to 500 nts, all believed to fold into a unique documented conformation. 

\begin{figure}
	\includegraphics[width=\linewidth]{graphs/RsampleVsIPANEMAP/Accuracy}
	\caption{\Rev{Comparison of predictive performances (GM) achieved by  \OurTool{} and \Software{Rsample} over the Hajdin\etal\cite{Hajdin2013} data set, consisting of 24 RNAs which associated SHAPE reactivities. The length of each individual RNA is indicated within square brackets.} \label{fig:Vsrsample}}
\end{figure}

\Rev{We used the default number of sampled structures for \Software{RSample}, namely 10~000 to compute correction factors and 1~000 for the clustering phase, and \OurTool (1~000 structures).  For all RNAs, a single structure was returned by both software. Note that the Pareto-optimality always implies a single returned cluster/structure for \OurTool{} in a mono probing setting.}
We computed and report in Figure~\ref{fig:Vsrsample} the individual Geometric Means (GM) associated with predicted structures. Sensitivity and PPV reported in Supp. Table S4, while Supp. Table S5 reports the stability of predictions across 10 runs of \OurTool.


%\paragraph{Results description}
A first observation is that \OurTool generally outperforms \Software{RSample}, averaging $80\%$ GM as opposed to $68\%$ for its competitor. \Rev{The superiority on \OurTool on this benchmark appears statistically significant according to a two-tailed paired $t$-test (P-value$=6\times10^{-3}$) with observed data.}
Out of the 24 considered RNAs, \OurTool outperforms its competitor for 16 out of the 24 RNAs, achieves similar performances for 2 of them, and is dominated for the remaining 6 cases. Whenever dominant, \OurTool achieves an average GM score of $82\%$, dominating by $32.8\%$ its competitor, while being dominated by $3.5\%$ over the remaining examples (including ties). 
%Difference<-c(IPANEMAP-Rsample)
%> sum(Difference[1:18])
%[1] 3.17
%> sum(Difference[19:24])
%[1] -0.382
Remarkably, \OurTool predicts secondary structures much more consistently than its competitor. 
It ensures GM that are greater than 50\% in all cases, and even exceeds 60\% over all except for the yeast tRNA, encoding the Aspartic acid and HIV1 5' pseudoknot domain. The standard deviation of GM for \OurTool is $12.4\%$, compared to the larger value of $21\%$ for \Software{RSample} predictions.


%\paragraph{Discussion}



This observed discrepancy is somewhat surprising, given that \OurTool and \Software{RSample} share the same sampling/clustering based modeling. However, the two methods differ substantially whenever a single conformation is detected, in which case \Software{RSample} switches to a simple MEA prediction, whereas \OurTool{} always proceeds with an iterative clustering. This allows \OurTool{} to consider a larger number of clusters, which we believe helps eliminating structural outliers, leading to the election of a better final centroid structure. 


%\clearpage


\subsection*{Comparing and exploiting multiple probing conditions: A case study on the Lariat capping ribozyme}


\begin{figure*}
	{\centering {\includegraphics[width=.9\linewidth]{graphs/didy/didy}}\\}
	
	\caption{\Rev{Model of the Lariat capping ribozyme from {\itshape Didymium iridis} (A -- PDB 4P8Z:A by Meyer\etal\cite{Meyer2014}), secondary structure, as annotated by \Software{DSSR}~\cite{Lu2015} (B) and \OurTool first-ranking prediction (C.), obtained from the integration of 8 maximally-diverging probing conditions (\NMIAMg, \OneMSevILUThreeMg, \NAIMg, \OneMSevCE, \CMCTMg,\OneMSevILUThree, and \OneMSevMgCE).}\label{fig:didy}}
\end{figure*}
To assess the potential of our method, we considered the Lariat capping ribozyme from \textit{Didymium Iridis} (\didy{}) whose 3D structure has been modeled at 3.85 \AA{} from X-ray diffraction data~\cite{Meyer2014}. \Rev{This RNA contains a large diversity of interactions, including helical domains, tertiary pairings, a kissing loop and a pseudoknot (Figure~\ref{fig:didy}). Its automated modeling is therefore a challenging case for computational  structure prediction methods.  For instance, running \Software{RNAfold} with default parameters only recovers four (DP2, DP2.1, P10 and P9) of the ten helices. Those are universally recovered by energy-based prediction tools, so differences in performances will mainly be observed between nucleotides 35 and 163.}

As described in the Material and Methods section, we probed \didy{}  in 16 different conditions using different SHAPE and classical reagents, in presence/absence of Mg$^\text{2+}$ ions, using either the standard premature stop or the newly developed mutation induction technologies to map the modification sites. These experimental results were then used as input of \OurTool{}, either individually or combined,  for secondary structure prediction. 


\paragraph{Comparison and clustering of conditions.}
Faced with diverse probing conditions, we first assessed the compatibility of the conclusions drawn from different probing data, including the probing-free run of \OurTool as a control. We used the methodology described in the Material and Methods~\ref{sec:dotplots} and, for each condition, computed the base pair probability distribution (aka \emph{dot-plot}) in the  pseudo-Boltzmann ensemble induced by the reactivity profile. We then computed the Ensemble Distance, the squared Euclidean distance between dot plots, for each pair of conditions. Figure~\ref{fig:PCA} summarizes the pairwise distance, projected onto a 2D surface by a principal component analysis.


\begin{figure}
	{\centering \includegraphics[width=\linewidth]{graphs/didy/PCA}\\}%, trim= 3.5cm .5cm 1cm 0,clip]
	
	\caption{2D spatial representation of conditions, computed by Principal Component Analysis (PCA) to optimally reflect pairwise ensemble distances between conditions. Colors and grayed areas indicate clusters of conditions.}\label{fig:PCA}
\end{figure}

A visual inspection of Figure~\ref{fig:PCA} suggests the presence of 8 clusters. In order to objectively build groups of compatible conditions, we performed a k-mean clustering using \Software{scikit-learn}, setting k to 8, and obtained the clusters highlighted in Figures~\ref{fig:biclustering} and \ref{fig:PCA}.  We obtain the following clusters:
\begin{bulletlist}%\setlength{\itemsep}{0pt}\setlength{\parskip}{.1em}	
\item \BullLab{A}: \OneMSevILU, \NMIA, and \NMIAMg;
\item \BullLab{B}: \OneMSevCE, \NMIACE, \BzCN, and \NAICE;
\item \BullLab{C}: \OneMSevILUThreeMg and \OneMSevILUMg;
\item \BullLab{D}: \OneMSevILUThree;
\item \BullLab{E}: \NMIAMgCE and \OneMSevMgCE;
\item \BullLab{F}: \DMSMg and probing-free;
\item \BullLab{G}: \CMCTMg and \NAIMg;
\item \BullLab{H}: \BzCNMg.
\end{bulletlist}
The predicted clusters are generally consistent with the ordering resulting from a spectral biclustering, implemented in \Software{scikit-learn} and executed with default parameters, as illustrated by \Rev{Figure~\ref{fig:biclustering}}. 

The average ensemble distance and PCA visualization support a status of outliers for conditions in the cluster \Bull{B} (\OneMSevCE, \NMIACE, \BzCN, and \NAICE).  We hypothesize that such conditions may be either representative of alternative conformations, or be altogether erroneous. Consequently, we will single out this cluster within our detailed analysis of the multi-probing performances.

%\bsi{sauf erreur de ma part le cluster B des parias c'est tous les CE sans Mg .je me demande s'ils n'auraient pas tous un même biais expérimental...}






\begin{figure}
	{\centering \includegraphics[width=\linewidth]{graphs/didy/bi_clustering}\\}%, trim= 3.5cm .5cm 1cm 0,clip]
	
	\caption{Discretized ensemble distance induced by conditions. Conditions ordered by spectral biclustering. Conditions regrouped within clusters by k-mean clustering appear as blocks on the x and y axis.}\label{fig:biclustering}
\end{figure}






%\clearpage

\paragraph{Assessment of predictions informed by individual probing conditions (mono probing).}
%First, we assess the individual performance of each probing condition considered as the sole source of structural constraints.  Then, analyze outperforming experimental conditions that ensure more accurate predictions, and spot \Def{outlier} conditions which, on the contrary, seem to increase the noise level, consequently leading to lower prediction performances for \OurTool{}. 



For each probing condition, we executed \OurTool{} on a single reactivity profile, using a sample size of $1~000$ structures. For the sake of reproducibility, we executed \OurTool{} $10$ times for each condition, and report in Table~\ref{tab:thermo-ipan} the average MCC values. We also report for reference the MCCs obtained by running \Software{RNAfold} with default parameters in energy minimization (MFE) and Maximum Expected Accuracy (MEA) modes.


The predictive performances of \OurTool averages a 70.5\% MCC, compared to 68.5\% and 68\% average MCC for MEA and MFE predictions respectively.
\Rev{This suggests modest, barely significant, improvements compared to classic prediction paradigms (P-values of 2\% and 14\% respectively).}
The stability of predictions across conditions seem comparable for all \OurTool, MEA and MFE-driven predictions (std dev. of 8\%). Excluding the outliers of cluster \Bull{B} improved the average MCC to 74\%.

Across the sixteen conditions, we observed a large discrepancy in the capacity of the experimental setup to inform predictions, with MCCs ranging from above 80\% (\NMIA, \NMIAMg, \OneMSevILU and \OneMSevILUMg) to around 60\% (\NAICE, \BzCN, \OneMSevCE and \NMIACE). \Rev{Interestingly the MCC value cannot be trivially anticipated from the compatibility of the reactivity scores with the predicted structure, or even with the native structure (see Supp. Table S6).}
 These predictive capacities clearly outperformed those achieved in the absence of probing data (51\% MCC), and can be interpreted as indicative of predictions of good overall quality. Indeed, it is worth reminding that the MCC is a stringent metric, taking values between -100\% and 100\%, 0\% being the expected MCC of a "coin tossing" random predictor. The lowest MCC value of 60\% is, in particular, equally consistent with Sensitivity/PPV values of 60\%/60\%, 80\%/46\% or 45\%/82\%.

Unsurprisingly, the presence/absence of Magnesium ions during the probing was observed to impact the predictions, with an observed drop from an average 73\% MCC in the presence of Mg$^\text{2+}$ to 67\% MCC in the absence of Mg$^\text{2+}$. However, while the poorly performing conditions in Cluster \Bull{B} \Rev{were} probed in the absence of Mg$^\text{2+}$, some of the most informative profiles (\OneMSevILU and \NMIA) were obtained in absence of Mg$^\text{2+}$.
\Rev{Even more drastic changes of performances were observed when comparing the MCCs of SHAPE-MaP experiments (avg 78\%) with the historical SHAPE CE technology (avg 66\%).}

\begin{table}
	\Heatset{min=60,   
		max=83,   
		max colour=Aquamarine3, % colour at maximum
		min colour=BadCol,      % colour at minimum
		Min colour=BadCol, % colour for values below min
		Max colour=Aquamarine3   % colour for values above max
	}
	\newcommand{\M}[1]{\multicolumn{1}{l}{#1}}
	
	{\centering
			\begin{adjustbox}{max width=\linewidth}
		\begin{tabular}{@{}lccHHH}
			\toprule
			                            &          &           & \multicolumn{3}{@{}c@{}}{MCC\%}              \\
			Condition                   &  Clust.  & Mg$^{2+}$ & \M{\relsize{-1}\OurTool} & \M{MFE} & \M{MEA} \\ \midrule
			\OneMSevILUMg               & \Bull{C} &    \Mg    & 85                       & 82      & 82      \\
			\OneMSevILU                 & \Bull{A} &   \NoMg   & 84                       & 82      & 84      \\
			\NMIAMg                     & \Bull{A} &    \Mg    & 81                       & 80      & 80      \\
			\NMIA                       & \Bull{A} &   \NoMg   & 80                       & 69      & 69      \\
			\OneMSevILUThreeMg          & \Bull{C} &    \Mg    & 74                       & 75      & 75      \\
			\NMIAMgCE                   & \Bull{E} &    \Mg    & 73                       & 72      & 73      \\
			\NAIMg                      & \Bull{G} &    \Mg    & 73                       & 68      & 70      \\
			\BzCNMg                     & \Bull{H} &    \Mg    & 71                       & 68      & 54      \\
			\OneMSevMgCE                & \Bull{E} &    \Mg    & 70                       & 69      & 70      \\
			\CMCTMg                     & \Bull{G} &    \Mg    & 70                       & 68      & 73      \\
			\OneMSevILUThree            & \Bull{D} &   \NoMg   & 64                       & 56      & 61      \\
			\DMSMg                      & \Bull{F} &    \Mg    & 63                       & 66      & 67      \\
			\NMIACE                     & \Bull{B} &   \NoMg   & 61                       & 60      & 60      \\
			\OneMSevCE                  & \Bull{B} &   \NoMg   & 60                       & 61      & 61      \\
			\BzCN                       & \Bull{B} &   \NoMg   & 60                       & 59      & 59      \\
			\NAICE                      & \Bull{B} &   \NoMg   & 60                       & 59      & 59      \\ \midrule
			Avg Technology              &          &  $-$  & 78                       & 74      & 75      \\
			                            &          &  $-$  & 66                       & 65      & 65      \\
			Avg -/+ Mg$^{2+}$           &          &   \NoMg   & 67                       & 64      & 65      \\
			                            &          &    \Mg    & 73                       & 72      & 71.5    \\ \midrule
			{\bfseries Average overall} &          &  $-$  & 70.5                     & 68      & 68.5    \\ \bottomrule
			                            &          &           &                          &
		\end{tabular}
		\end{adjustbox}}
	
	%data are from multiprobing7 
	\caption{\Rev{Predictive performances (MCC) of \OurTool{}, averaged over 10 runs, for the \didy{} dataset, consisting of 16 reactivity profiles. (Pseudo) MFE and MEA structures, predicted using \Software{RNAfold}~\cite{Lorenz2011}, are included for reference. The structure was probed in presence (\Mg) or absence (\NoMg) of Mg$^{2+}$.}
	}
	\label{tab:thermo-ipan}
\end{table}




\begin{figure}
	\newcommand{\W}{.7}
	{\centering\includegraphics[width=\W\linewidth]{graphs/didy/pairs-spiders}\\}
	\caption{Shift in \OurTool{} predictive capacity (MCC) observed upon considering two conditions, compared to the worst ({\sf\bfseries A}),  average ({\sf\bfseries B}) 
		and best ({\sf\bfseries C}) 
		predictions achieved for mono-probing conditions. Green and red edges indicate enhanced and degraded predictions respectively compared to the reference (Min% 
		, Max 
		or Average of conditions). Thickness indicates absolute shift value (only values above 5\% reported for readability).\label{fig:pairwise}}
\end{figure}

\paragraph{Bi-probing mitigates the influence of outliers, but does not significantly improve prediction quality.}
Next, we turned to a systematic exploration of the predictive capacities of bi-probing analyses, based on pairs of probing profiles, and attempted to quantify their impact on \OurTool predictions. For each pair and triplets of conditions, we executed \OurTool using 1~000 structures per condition, and considered the first returned structure. We then computed the associated MCC, and compared it with the MCC of the worst performing condition (Min), best-performing condition (Max) and average MCC over the single conditions experiments. A summary of the results over pairs is shown in Figure~\ref{fig:pairwise} (see details in Supp. Table S7). 

A superficial inspection of the resulting MCCs appears to confirm the conclusions reached on the Cordero\etal\cite{Cordero2012} dataset. Indeed, considering two conditions led to predictions whose quality fall between the worst and the best one, while being generally close to the average. More precisely, the average MCC value of pair-informed predictions remained at 70.7\%, improving only by 0.2\% over the average MCC of mono-probing predictions. Pair-informed predictions were marginally more reliable, with the median MCC \Rev{increasing} to 72.5\% MCC from 70.5\% for mono-probing predictions. However, outliers strongly impacted the overall picture, and ignoring the conditions in cluster \Bull{B} increased the average MCC to 75.1\%, while unsurprisingly decreasing the standard deviation (8.6\% $\to$ 6.5\%).

Compared to the minimum MCC of the pair, the MCC of bi-probing predictions was improved by 5.3\% on average over the MCC of the worst-performing conditions. Pair MCCs exceeded their associated min. MCC by at least 5\% in 47 pairs out of the 120 possible pairs, while never being dominated by more than  5\% MCC. 

Bi-probing seemed to perform similarly as the average of the two mono-probing conditions, with an average MCC improvement of 0.2\%. However, for 56 out of the 120 possible pairs, the bi-probing conditions exhibited an improvement of at least 1\% over the average, while a decay of at least 1\% was observed for 46 out of the remaining 64 pairs. Such a decay can partly be attributed to the disruptive effect of the outliers from cluster \Bull{B}, involved in 35 of the 46 observed loss of predictive performance. Moreover, removing this cluster induced an average MCC improvement of 1.1\%.

Predictions informed by two conditions remained, however, generally dominated by the best performing condition of the pair. On average, the MCC of the best condition is 4.9\% higher than the one achieved by bi-probing analysis. 
Outliers in cluster \Bull{B} are partly responsible for this situation, and their removal reduces the average MCC decay to 3.3\%. Moreover, for 9 pairs of conditions, the bi-probing analysis produced MCCs that are at least 1\% better than the best of its two conditions, but these encouraging examples were generally dominated by 66 examples where the pair performs more than 1\% worse than the best mono-probing condition.

\paragraph{Considering three conditions yields improvement over the average performance of the triplet.}
Increasing the number of conditions to three boosted the average MCC to 72.3\% and even reached 76.5\% over triplets that do not contain conditions from cluster \Bull{B}. The MCC gain over the worst condition in the triplet reached 9.1\% on average.  Predictions were also more consistently good, with a median MCC of \%73.4. Disregarding conditions in cluster \Bull{B} did stabilize the quality of prediction (stdev 8.4\% $\to$ 5.5\%), while increasing the average MCC to 77.4\%.

Compared to bi-probing, MCC values achieved by integrating triplets of conditions indicated a clear gain over the average performances of individual conditions, supporting a notion of complementary between experiments. Indeed tri-probing experiments induced an average +1.7\% MCC improvement (median +2.6) over the average of mono-probing prediction in the triplet \Rev{(P-value$=1.4\times 10^{-9}$)}. The gain of quality was widespread, and observed for 65\% of the triplets. Finally, 39 triplets were found to induce MCC values greater than 83\%.

The enhanced performances of tri-probing over bi-probing \Rev{was deemed statistically significant (P-value$=2\times 10^{-2}$), and} consistent with the \emph{voting} principle underlying the clustering used in \OurTool. \Rev{Indeed, in a bi-probing setting, a single outlier (\emph{e.g.} \NAICE) may entirely determine the final structure due to its Boltzmann ensemble being very tightly concentrated around a, presumably erroneous, centroid structure. In the presence of three or more conditions, however, clusters resulting from an outlier condition are typically expected to be dominated in the Pareto front, by compatible structures originating from alternative conditions. It follows that the influence of outliers over the final prediction is mitigated.}

%\clearpage


\begin{table}[ht]
	\Heatset{min=60,   
		max=83,   
		max colour=Aquamarine3, % colour at maximum
		min colour=BadCol,      % colour at minimum
		Min colour=BadCol, % colour for values below min
		Max colour=Aquamarine3   % colour for values above max
	}
    \newcommand{\Q}{\relsize{-1}}

	{\centering
		\begin{adjustbox}{max width=\linewidth}
			\begin{tabular}{@{}llHH@{}}
				\toprule
				&                                           & \multicolumn{2}{@{}c@{}}{MCC\%}                     \\
				& Conditions                                & \multicolumn{1}{@{}c@{}}{\Q{}Joint} &  \multicolumn{1}{@{}c@{}}{\Q{}Mono} \\ \midrule
				{Cluster \Bull{A}} & \NMIAMg, \OneMSevILU, \NMIA                       & 80                  & 82                      \\
				& \NMIAMg, \OneMSevILU                             & 82                  & 82                      \\
				& \OneMSevILU, \NMIA                               & 82                  & 82                      \\
				& \NMIAMg,  \NMIA                              & 80                  & 80                      \\ \midrule
				{Cluster \Bull{B}}                         & \NAICE, \BzCN, \OneMSevCE, \NMIACE                   & 60                  & 60                      \\
				& \NAICE, \BzCN, \OneMSevCE                           & 60                  & 60                      \\
				& \NAICE, \BzCN, \NMIACE                          & 61                  & 60                      \\
				& \NAICE, \OneMSevCE, \NMIACE                         & 61                  & 60                      \\
				& \BzCN, \OneMSevCE, \NMIACE                        & 61                  & 60                      \\
				& \NAICE, \BzCN                                  & 60                  & 60                      \\
				& \NAICE, \OneMSevCE                                 & 59                  & 60                      \\
				& \NAICE, \NMIACE                                & 60                  & 60                      \\
				& \BzCN, \OneMSevCE                                & 61                  & 60                      \\
				& \BzCN, \NMIACE                                & 60                  & 60                      \\
				& \OneMSevCE, \NMIACE                              & 62                  & 61                      \\ \midrule
				{Cluster \Bull{C}}                         & \OneMSevILUMg, \OneMSevILUThreeMg                        & 82                  & 80                      \\ \midrule
				{Cluster \Bull{E}} & \OneMSevMgCE, \NMIAMgCE                          & 73                  & 72                      \\ \midrule
				{Cluster \Bull{G}}                         & \NAIMg, \CMCTMg                              & 73                  & 72                      \\ \bottomrule
				%				\midrule
				%				{All 16 conditions} & .80                   & .70                       \\ \bottomrule
			\end{tabular}
		\end{adjustbox}
		\\}
	\caption{Predictions do not benefits from the joint consideration of similar conditions. For each cluster, all subsets of conditions in the cluster are considered, and the MCC of the resulting prediction ('Joint') is compared to the average MCC of predictions performed with individual conditions independently ('Mono'), revealing little improvement.}\label{fig:similarconds}
\end{table}

\paragraph{Similar conditions do not contribute to the quality of predictions.} 
Our distance assessment and clustering on conditions revealed groups of conditions that were highly compatible in their conclusions, while others seemed to include highly diverging structural information. While some outlier conditions, such as those of cluster \Bull{B}, clearly stood out as erroneous, the results of our systematic analysis of pairs and triplets supported a notion of \Def{complementarity between conditions}. Under this assumption, downstream analyses would be more likely to benefit from the presence of highly diverging probing experiments, than the accumulation of similar (arguably redundant) profiles. 

To test this hypothesis, we executed \OurTool{} on every subset of conditions within clusters, and report in Table~\ref{fig:similarconds} the quality (MCC) of predictions for any subset of similar conditions (see Supp. Figure~S1 for associated models). 
\Rev{Note that such an analysis does not use knowledge of the native structure, since our clustering only relies on properties of the pseudo-Boltzmann ensemble.}
 
As expected, compared to the average MCC over associated mono-probing analyses, a joint consideration of similar conditions only induced limited progress over single probing analyses (between -1\% and +1\%), leading to an modest average improvement (+0.2\% average MCC). This is consistent with the notion that supplementing a probing-based modeling with compatible conditions provides very little additional information, leading to a limited contribution of similar conditions. 



\paragraph{Considering diverse conditions improves and stabilizes the quality of predictions.} 
Having established the redundant nature of similar conditions, and their limited contribution to downstream modeling, we turn to an analysis of conditions across clusters.  We executed \OurTool{} to assess the quality of predictions informed by conditions chosen across the 8 clusters. We considered any possible combination, except for the outlier cluster~\Bull{B}, for which we only considered \NAICE as a representative.

The results, summarized in Table~\ref{fig:differentconds}, demonstrate a clear advantage of including diverse conditions, with an average MCC value of 77\% in the absence of condition from cluster \Bull{B}. Those performances compare favorably against the 72\% MCC achieved on average by individual conditions \Rev{(P-value$=3\times 10^{-5}$)}, and reveal a remarkable stability (5\% stdev), and a degradation in only two out the 24 tested cases ($\sim$1\% \Rev{loss of MCC} in both cases).  A similar trend can be observed in the disruptive presence of \NAICE, inducing an 72\% average MCC, albeit with an increased standard deviation of 9\%.

Finally, considering the 16 conditions together induced a 80\% MCC value, increasing to 83\% in the absence of cluster \Bull{B}. \Rev{Overall, these results support the notion that the strategy used} by \OurTool{} is able to exploit the complementary information provided by multiple conditions, contributing to better predictions.

	\begin{table}[h]
	\Heatset{min=60,   
		max=83,   
		max colour=Aquamarine3, % colour at maximum
		min colour=BadCol,      % colour at minimum
		Min colour=BadCol, % colour for values below min
		Max colour=Aquamarine3   % colour for values above max
	}
    \newcommand{\MC}[1]{\multicolumn{1}{@{}c@{}}{#1}}
    \newcommand{\MCB}[1]{\multicolumn{2}{@{}c@{}}{#1}}
    \newcommand{\Q}{\relsize{-1}}

	\begin{adjustbox}{max width=\linewidth}
		\begin{tabular}{@{}llllHHHH@{}}
			\toprule
			\multicolumn{4}{@{}l@{}}{}                                    & \multicolumn{4}{@{}c@{}}{MCC\%}\\
			\multicolumn{4}{@{}l@{}}{Conditions in individual clusters}                                    & \MCB{-NAI} & \MCB{+NAI} \\
			\Bull{A} & \Bull{C} & \Bull{E} & \Bull{G}   & \MC{\Q Joint} & \MC{\Q Mono}  & \MC{\Q Joint}       & \MC{\Q Mono}      \\ \midrule
			\Q \NMIAMg          & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \NAIMg             & 84       & 72       & 83             & 71           \\
			\Q \NMIAMg          & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \CMCTMg            & 75       & 72       & 82             & 71           \\
			\Q \NMIAMg          & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \NAIMg             & 79       & 73       & 82             & 71           \\
			\Q \NMIAMg          & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \CMCTMg            & 82       & 72       & 60             & 71           \\
			\Q \NMIAMg          & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \NAIMg             & 70       & 71       & 60             & 70           \\
			\Q \NMIAMg          & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \CMCTMg            & 70       & 70       & 72             & 69           \\
			\Q \NMIAMg          & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \NAIMg             & 72       & 71       & 74             & 70           \\
			\Q \NMIAMg          & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \CMCTMg            & 75       & 71       & 60             & 70           \\
			\Q \OneMSevILU          & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \NAIMg             & 82       & 73       & 75             & 71           \\
			\Q \OneMSevILU          & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \CMCTMg            & 84       & 72       & 60             & 71           \\
			\Q \OneMSevILU          & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \NAIMg             & 73       & 73       & 60             & 72           \\
			\Q \OneMSevILU          & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \CMCTMg            & 82       & 73       & 82             & 71           \\
			\Q \OneMSevILU          & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \NAIMg             & 84       & 71       & 60             & 70           \\
			\Q \OneMSevILU          & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \CMCTMg            & 70       & 71       & 84             & 70           \\
			\Q \OneMSevILU          & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \NAIMg             & 74       & 72       & 60             & 70           \\
			\Q \OneMSevILU          & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \CMCTMg            & 74       & 71       & 75             & 70           \\
			\Q \NMIA            & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \NAIMg             & 82       & 72       & 75             & 71           \\
			\Q \NMIA            & \Q \OneMSevILUMg        & \Q \OneMSevMgCE          & \Q \CMCTMg            & 83       & 72       & 82             & 70           \\
			\Q \NMIA            & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \NAIMg             & 73       & 73       & 59             & 71           \\
			\Q \NMIA            & \Q \OneMSevILUMg        & \Q \NMIAMgCE         & \Q \CMCTMg            & 82       & 72       & 82             & 71           \\
			\Q \NMIA            & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \NAIMg             & 73       & 71       & 82             & 69           \\
			\Q \NMIA            & \Q \OneMSevILUThreeMg       & \Q \OneMSevMgCE          & \Q \CMCTMg            & 70       & 70       & 73             & 69           \\
			\Q \NMIA            & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \NAIMg             & 74       & 71       & 68             & 70           \\
			\Q \NMIA            & \Q \OneMSevILUThreeMg       & \Q \NMIAMgCE         & \Q \CMCTMg            & 74       & 71       & 73             & 69           \\ \midrule
			                &                 &                  & Average           & 77       & 72       & 72             & 70           \\ \bottomrule
		\end{tabular}
	\end{adjustbox}\\
	\caption{Predictive performances of conditions across clusters, coupled with impact assessment of NAI.  For each experiment, a single condition is chosen from each cluster except for cluster \Bull{B}. Clusters \Bull{D}, \Bull{F} and \Bull{H}, each consisting of a single condition, contribute their sole condition, leading to the list of conditions  $\mathcal{M}$:=\{\OneMSevILUThree, \BzCNMg and \DMSMg\}. 
		For each combination, the MCC of the predicted structural model is reported, as predicted from the conditions in the presence/absence of \NAICE ('Joint' columns). For each prediction, the average MCC value of mono conditions is reported as a baseline ('Mono' columns).
	}\label{fig:differentconds}
\end{table}

%
%	\begin{table}
%	\newcommand{\A}{45}
%	\newcommand{\B}{1.2em}
%	\newcolumntype{C}[1]{>{\centering\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
%
%	\centering
%	\begin{adjustbox}{max width=\linewidth}
%		\begin{tabular}{@{}>{\centering}p{\B}>{\centering}p{\B}>{\centering}p{\B}>{\centering}p{\B}>{\centering}p{\B}>{\centering}p{\B}>{\centering}p{\B}cc@{}}
%			\toprule
%			\rotatebox{\A}{\Bull{C} \OneMSevILUMg} & \rotatebox{\A}{\Bull{D} \OneMSevILUThree} & \rotatebox{\A}{\Bull{A} \OneMSevILU} & \rotatebox{\A}{\Bull{F} \DMSMg} & \rotatebox{\A}{\Bull{E} \NMIAMgCE} & \rotatebox{\A}{\Bull{G} \CMCTMg} & \rotatebox{\A}{\Bull{H} \BzCNMg} &      MCC      & Mono-Avg \\ \midrule
%			$\bullet$             &                                  &            $\bullet$            &           $\bullet$            &                                   &                                 &                                 & \textbf{.853} &   .77    \\
%			$\bullet$             &                                  &            $\bullet$            &                                &             $\bullet$             &                                 &                                 & \textbf{.853} &   .81    \\
%			$\bullet$             &                                  &                                 &           $\bullet$            &                                   &            $\bullet$            &                                 & \textbf{.853} &   .73    \\
%			$\bullet$             &            $\bullet$             &                                 &                                &                                   &                                 &            $\bullet$            & \textbf{.853} &   .73    \\ \bottomrule
%		\end{tabular}
%	\end{adjustbox}\\
%
%	\caption{Best triplets of conditions, achieving maximal obtained MCC value.}\label{fig:best}
%\end{table}
%\begin{table}
%	\centering
%	\begin{adjustbox}{max width=.5\textwidth}
%		\begin{tabular}{@{}llllll@{}}
%			\toprule
%			Description & MCC & Mono-Avg\\
%			\midrule
%			12 conditions& {\bfseries.83}&.74 \\
%			{\em All 16 conditions} &	\textbf{.8}& .705\\
%			&	\textbf{.6}& \\
%			\bottomrule
%		\end{tabular}
%	\end{adjustbox}\\
%
%	\caption{Report the best possible score; + all conditions}
%\end{table}
%%

%\begin{figure*}
%\begin{adjustbox}{max width=\linewidth}
%  \begin{tabular}{ll}
% Seq & \tt {GGUUGGGUUGGGAAGUAUCAUGGCUAAUCACCAUGAUGCAAUCGGGUUGAACACUUAAUUGGGUUAAAACGGUGGGGGACGAUCCCGUAACAUCCGUCCUAACGGCGACAGACUGCACGGCCCUGCCUCUUAGGUGUGUUCAAUGAACAGUCGUUCCGAAAGGAAGCAUCCGGUAUCCCAAGACAAUC
%} \\
% 
% NATIVE &  \tt {(((((..(((((..(((((((((.......)))))))))..((((.(((.((((......(((((...((((..(((......))).......))))((....)).............))))).(((.....))))))).)))..........((((....))))....))))...)))))..)))))} \\
%
%.865 &\tt{(((((..((((((.(((((((((.......))))))))).((((((((((((((......((((....((((...((.......)).......))))......(.(........)..).)))).(((.....)))))))))))..........((((....))))...)))))).))))))..)))))}\\
% \\
%\end{tabular}
%\end{adjustbox}
%\caption{Content: 3D didymium + native structure + predicted base pairs}
%\end{figure*}
%\begin{figure}
%\includegraphics[width=\linewidth]{graphs/Arcs}
%\end{figure}
%\subsection*{\OurTool{} yields reproducible predictions}
%\as{Which structure?}



\subsection*{Supplementing wild-type profiles with random single mutants increases prediction quality}
\Rev{In this final analysis, we tested the capacity of \OurTool to extract and exploit complementary information produced by the systematic probing of single-point mutants using the Mutate-and-Map (MaM) protocol~\cite{Cordero2015}.
The joint analysis of several mutants may appear ill-advised, as single point mutants are expected to adopt different structures than the wild-type sequence (WT) , an expectation that is at the core of downstream analyses~\cite{Cordero2015}. However, those changes are typically local, and the limited influence of outliers on \OurTool{} substantiates the hope that the benefits of including independently-produced conditions will outweigh the cost of noise, introduced by local changes.}

We generated 100 uniformly-distributed subsets consisting of 1, 2, 3 and 10 mutants extracted from the Cheng dataset~\cite{Cheng2017}, which we each supplemented with the WT sequence.
For each set of probing profiles (\Cond{DMS}{mg}{MaP}{ilu}{}), we executed \OurTool{} with default parameters, using a sample size of 1~000 structures. 
We also analyzed the WT sequence alone, reproducing the analysis 100 times to compare the variability across sets of mutants to the one induced by stochastic aspects of our method. 
Finally, we performed a restricted analysis focusing on the WT, supplemented with 2 probing profiles selected from the 20 most similar conditions to the WT , as assessed by the ensemble distance metric. By comparing our results with those obtained for unrestricted pairs of mutants, we assess the influence of structure variability on the quality of our predictions. We report in Figure~\ref{fig:variantanalysis} the distribution of MCC values associated with the first prediction returned by \OurTool (see Supp. Figure S2 for details).

For the WT alone, the predictions of \OurTool were associated with 61.8\% MCC, comparable to the worst condition of our previous dataset.
Our predictions were generally stable and, in 94/100 runs, resulted in a structure having MCC between 60.5\% and 61\%. The remaining 6 runs showed improved MCC values, ranging from 76\% to 78\% and suggesting the existence of an alternative conformation in the pseudo-Boltzmann ensemble.
%%\as{ This value can not be fixed since we have  0.6134: 12\%, 0.7597: 2\%, 0.7649: 1\%, 0.7783: 1\%, 0.7665: 1\%, 0.7531: 1\% , is it ok Yann not to specify the improvemnt percentage ?}
%% Counter({0.6084: 82, 0.6134: 12, 0.7597: 2, 0.7649: 1, 0.7783: 1, 0.7665: 1, 0.7531: 1})
%mean(Multiprobing_100WT) 

When the WT and a single random mutant (WT+1M) were jointly considered, the dispersion of the MCCs increased, and values ranging from 38.9\% to 85\% were observed.%min(Multiprobing_1M)
\Rev{The average MCC in the presence of a single mutant increased to 64.9\%,%mean(Multiprobing_1M)
suggesting a significant positive contribution from the additional mutant profile (P-value$=1\times 10^{-3}$).}

This trend is confirmed when two random mutants were considered in addition to the WT (WT+2M), with an average MCC across runs increasing to 66.6\%. %mean(Multiprobing_2M).
Interestingly, the dispersion of the MCC was lower for WT+2M than for WT+1M, with lower MCC values above 57\%.  This results suggest  a role of \emph{tie-breaker} for the third condition, allowing to mitigate the deleterious effect of some outlier.
However, outliers still played a disruptive role, and restricting chosen mutants to the 20 reactivity profiles most similar to the WT further improved the average MCC 2.8\% to 67.7\%,  \Rev{ a significant improvement compared to WT+1M (P-value$=2.4\times10^{-2}$).}

The average MCC again increased in the presence of 3 mutants, to reach an average MCC of 69.3\%, \Rev{ a substantial improvement that nevertheless fails to reach the level of statistical significance (P-value$=11\times 10^{-2}$).} For 10 mutants, the average MCC and overall distribution remains highly similar to that obtained by considering three mutants, suggesting \Rev{a plateau in the performances}. Overall, these results support a notion of complementarity between the probing data produced across reasonably similar mutants, leading to a gradual decrease of the signal to noise ratio. The bimodal nature of the distribution also suggests the existence of two conformations, equally supported by the WT and the mutant profiles.

\begin{figure}
	{\centering
		\includegraphics[width=\linewidth]{graphs/combinationscale.png}\\}
		
	\caption{\Rev{Distributions of predictive accuracies (MCC) of structures predicted by \OurTool{}, informed by Mutate-and-Map data, for \didy{} (Cheng dataset~\cite{Cheng2017}). Predictions were repeated 100 times, informed both by the wild-type sequence and reactivity profile (WT), and with randomly-selected reactivity profiles associated with 1, 2, 3, or 10 single-point mutants ({\sf WT+$k$M}). Pairs of mutants restricted to the 20 most similar to the WT were also considered ({\sf WT+2M close}). } }\label{fig:variantanalysis}
\end{figure}


%\begin{figure}[h]
%\begin{center}
%\includegraphics[scale=.4]{graphs/WT+combin}
%\label{Vsrsample}
%\caption{Accuracy of prediction reported as the MCC when considering only the WT, considering one mutants along side the WT and considering 10 mutants in addition to the WT.}

%\end{center}
%\end{figure}




%------------------------------------------------------------------------------

%======================




\section*{Conclusion and discussion}

We introduce \OurTool{}, an integrative method for the prediction of secondary structure models from multiple probing profiles. Based on the simple premise that good structural models should be \Rev{thermodynamically} stable and supported by most probing experiments, it uses a combination of stochastic backtrack in the pseudo-Boltzmann ensemble, coupled with a structural clustering to elect dominant conformations. This strategy is fast, \Rev{generally stable}, and leads to predictions that clearly benefit from the availability of multiple probing profiles, as demonstrated by detailed analyses performed of four datasets. 


To the best of our knowledge, \OurTool currently represents the only available method for a joint automated consideration of multiple probing profiles. Surprisingly, it also appears to produce the most accurate secondary structure models when informed by a single condition, as demonstrated by our reanalysis on the Hajdin\etal\cite{Hajdin2013} benchmark. We attribute this unexpected result to the adoption of a sampling/clustering approach, \Rev{which appears to eliminate spurious outliers from the conformational space even when a single conformation dominates.}
\Rev{In this context, the good performance of \OurTool in comparison with \Software{RSample}~\cite{Spasic2017} may appear surprising. Indeed, while independently developed and used~\cite{Deforges2017}, both methods share a common sampling/clustering scheme. However, \Software{RSample}~\cite{Spasic2017} reverts to a MEA~\cite{Lu2009} computation whenever a single conformation is suspected. The MEA structure coincides with the centroid of the whole Boltzmann ensemble, suggesting that the increased performances of \OurTool could be at least partially attributed to the use of sampling, and to our adaptive choice of the number of clusters.}
Moreover, even in cases where a single dominant conformation is expected, considering multiple clusters seems to create more opportunities for discarding noisy random unstable conformations. Such conformations naturally occur in a Boltzmann-distributed subset of structures, and may end up \emph{polluting} the centroid structures by artificially inflating the cluster diversity. 
%The impact on \Software{RSample} predictions of such a \emph{noise} seems worsened by the fact that it returns the MEA, the centroid of the whole ensemble, whenever a dominant conformation is detected.



Our analyses reveal that integrating multiple profiles of at least three conditions allows to mitigate the effect of outlier conditions and, at times, to produce better secondary structure models than the one inferred only from the most accurate profile. A comparison of the joint performances to the average appears fair since no single reagent appears to systematically dominate in term of predictive capacity. It follows that a modeler cannot objectively favor \emph{a priori} a probing condition/reagent~\cite{Yu2018}. In other words, multiple experimental probing profiles, even combined in a pairwise fashion, \Rev{can be used jointly to mitigate the empirical risk of a misprediction}. 


\Rev{Current limitations of \OurTool include the lack of an explicit support for pseudoknots, which will be addressed in a future version, using  polynomial-time dynamic programming~\cite{Janssen2015}, or a iterative heuristic alternative~\cite{Hajdin2013}. The inherently stochastic nature of the sampling scheme may also be challenging for certain types of analysis. However, despite its stochastic foundations, \OurTool{} is typically stable in its predictions, as can be seen in Figure~\ref{fig:reproducibility} for all probing datasets produced over our \didy case study.
Interestingly, the most notable exception to the general stability is observed for the probing-free prediction, consistent with the widespread interpretation of SHAPE-induced pseudo-potentials as focusing predictions onto a subset of the Boltzmann ensemble. A similar behavior is observed for the other datasets considered in our study (see Supp. Tables S2 and S5). %In the interest of reproducibility, ipanemap reports, and accepts as an option, the random seed used both within the main software and its non-deterministic dependencies, leading to identical repeats of the experiments.
Finally, the current clustering method induce a growth of time and memory requirements that scales quadratically with the number of samples. While this allows a joint analysis of a dozen of reactivity profiles in less than an hour on a personal computer, we expect the ability to push the number of samples will lead to better, even more reproducible, predictions. To that purpose, we will explore embedding techniques and linear-time community detection algorithms as alternatives to MBkM~\cite{Sculley2010}.}



Future extensions also include \Rev{addressing} the dilemma of whether or not to include Mg$^{\text{2+}}$ ions during the probing, in the perspective of a downstream computational prediction. On the one hand, RNA reaches its native 3D structure in the presence of Mg$^{\text{2+}}$ ions~\cite{Knapp1989}. However, such native structures include non WC pairings, and complex topological motifs such as pseudoknots, kissing loops, tetraloop interactions, that classic structure prediction algorithms do not explicitly capture. In contrast, in the presence of only monovalent ions, RNA is expected to form its helical domains, but leave its  tertiary contacts unstable, with the notable exception of the G quadruplex. Therefore, probing the RNA without Mg$^{\text{2+}}$ could provide a signal that is less deceptive, and thus more informative, to predicting algorithms. Unfortunately, destabilizing tertiary contacts increase the probability of misfolding, or the coexistence in solution of multiple conformations. In this work we probed \didy, which contains many tertiary motifs, in the presence and absence of Mg$^{\text{2+}}$. Remarkably, the most accurate predictions were obtained with 1M7 and NMIA \Rev{irrespectively of the presence/absence of Mg$^{\text{2+}}$, and despite the fact that reactivity profiles substantially diverged between the two conditions (see Supp. Fig. S4)}. However, for other reagents/protocols, predictions appeared positively impacted (+$10\%$ average MCC) by the presence of Mg$^{\text{2+}}$, painting a mixed picture that would deserve future investigations.
%\item je pense qu'il serait intéressant au moins pour un uo deux réactif de voir ce que donne le couple + et - Mg. Peut etre par exemple avec le 1M7ilu et un autre mauvais genre le NAI
%\end{itemize}
% il faudrait regrder ce que fait weeks avec le 1M7, le NMIA et le 1M6, je n'ai pas bien compris quelle tambouille il fait, mais ca part du meme genre de constatation : il y a une spécificité dans chacune des sondes qu'il n'arrive pas vraiment à formaliser donc il fait une tambouille qui lui permet de tous les intégrer (mais c'est peut etre un peu pédestre comme démarche!). Apres ce sont probablement des conditions tres proches ce qui n'apporte pas d'apres nos résultats bcq d'infos supplémentaires .. bref ca vaudrait peut etre le coup de comparer.


\begin{figure}
	{\centering\includegraphics[width=1\linewidth]{graphs/boxplotreproducibility}\\}
	
	\caption{Stochastic variability of \OurTool{} predictions for \didy{} dataset. MCC of predicted structures over 10 runs of \OurTool{}, in presence/absence of Mg$^{2+}$, using both \Rev{SHAPE CE and MaP} protocols for a collection of reagents. }\label{fig:reproducibility}
\end{figure}




%
%%discussion et conclusion sont fusionnés dans ce type d'article.}
%
%
%
%\begin{itemize}
%	\item Our integrative approach compared to other tools did exhibit a trade-off  between  number of optimal clusters and accuracy but remains fast to execute. 
%	\item While based on the assumption that the different probing condition reveal complementary aspects of the same structure, it can still be enriched by experiments where the dominant conformation is expected to differ from the native one.
%	
%	\item 
%	
%	\item Complementarity
%	\item Pseudo-Energy models
%\end{itemize}

\section*{Authors contributions}
AS, BS, and YP designed the computational method. 
AS and YP implemented the computational method.
DA and BS planned and carried out all probing experiments.
AS and YP planned and carried out the computational experiments. 
AS, DA, BS and YP contributed to the interpretation of the results. 
All authors provided critical feedback and helped shape the research, analysis and manuscript

\section*{Acknowledgements}
The authors wish to express their gratitude to Ronny Lorenz for developing a probing-informed version of \Software{RNAsubopt}, allowing to sample the pseudo-Boltzmann ensemble, Nathalie Chamond for helpful suggestions and Benoit Masquida for the gift of \didy{} plasmid and helpful discussions about protocols and \didy{} structure .

\section*{Funding}
This work, including AS and DA \Rev{PhD} fellowships, has been supported by La Fondation pour la Recherche Medicale ([FRM
DBI20141423337]).
%\bibliographystyle{bioinformatics}
\bibliographystyle{nar}
%\bibliographystyle{natbib}%compa
\bibliography{biblio}

\end{document}
