\documentclass[11pt]{article} % use larger type; default would be 10pt

\usepackage{float}
\usepackage[utf8]{inputenc} % set input encoding (not needed with XeLaTeX)
\usepackage{adjustbox}   
%%% Examples of Article customizations
% These packages are optional, depending whether you want the features they provide.
% See the LaTeX Companion or other references for full information.
  
%%%%%%%%%%%%%%%%%%%%%%%%%%formatting cells colors
\usepackage[table,x11names,dvipsnames]{xcolor}
 \usepackage{collcell}
 \usepackage{array}
 \usepackage{tikz}
 \usepackage{relsize}
  \usepackage{pgfkeys}
 \usepackage{graphicx}
 \usepackage{xspace}
 \usepackage{amsmath,amssymb}

%%% PAGE DIMENSIONS
\usepackage{geometry} % to change the page dimensions
\geometry{a4paper} % or letterpaper (US) or a5paper or....
 \geometry{margin=1in} % for example, change the margins to 2 inches all round
% \geometry{landscape} % set up the page for landscape
%   read geometry.pdf for detailed page layout information
\usepackage{multicol}
\usepackage{booktabs}
\usepackage{todonotes}
\usepackage{times}
\usepackage{inconsolata}
\usepackage{subcaption}

\usepackage[figurename=Supplementary~Figure, tablename=Supplementary~Table]{caption}
\DeclareCaptionLabelFormat{cont}{Figure S\arabic{figure}}
\captionsetup[figure]{labelformat=cont}
\DeclareCaptionLabelFormat{contt}{Table S\arabic{table}}
\captionsetup[table]{labelformat=contt}
\captionsetup{labelfont={bf}}

\input{macros.tex}

\renewcommand{\Def}[1]{{\bf #1}}

\title{\OurTool{}:  Integrative Probing Analysis of Nucleic Acids Empowered by Multiple Accessibility Profiles\\[1em]
	Supplementary material
}
\author{}
\date{} % Activate to display a given date or no date (if empty),
         % otherwise the current date is printed 
\newcommand{\Start}{s}
\newcommand{\End}{e}


\begin{document}
\maketitle

\tableofcontents

\newpage

\section{Multiprobing benchmark (Cordero~\emph{et al}~\cite{Cordero2012} dataset)}
\subsection{GM values of individual predictions}

 \Heatset{min=0.4,  
	max=1,   
	max colour=Aquamarine3, % colour at maximum
	min colour=BadCol,      % colour at minimum
	Min colour=BadCol, % colour for values below min
	Max colour=Aquamarine3   % colour for values above max
}

\begin{table}[H]
\centering
\newcommand{\B}{\multicolumn{1}{c}{$\bullet$}}
\newcommand{\BR}{\multicolumn{1}{c}{$\bullet$}}
\newcommand{\D}{\multicolumn{1}{c}{\color{gray}$\circ$}}
\newcommand{\DR}{\multicolumn{1}{c}{\color{gray}$\circ$}}
\begin{adjustbox}{max width=1\textwidth}
 \begin{tabular}{@{}lHHHHHHHHHHHHHHHH@{}}
 	\toprule
 	                              & \multicolumn{8}{l}{ \OurTool{}}                                     & \multicolumn{4}{l}{\Software{RNAfold} -- MFE mode} & \multicolumn{4}{l}{\Software{RNAfold} -- MEA mode}  \\
 	\textbf{CMCT}                 & \D    & \B    & \D       & \D    & \D    & \B       & \B    & \BR   & \D       & \D    & \B    & \DR                     & \D    & \D    & \B       & \D                       \\
 	\textbf{NMIA}                 & \D    & \D    & \B       & \D    & \B    & \D       & \B    & \BR   & \D       & \D    & \D    & \BR                     & \D    & \D    & \D       & \B                       \\
 	\textbf{DMS}                  & \D    & \D    & \D       & \B    & \B    & \B       & \D    & \BR   & \D       & \B    & \D    & \DR                     & \D    & \B    & \D       & \D                       \\ \midrule
 	5SRNA,E.Coli                  & 0.25  & 0.238 & 0.247    & 0.244 & 0.247 & 0.247    & 0.254 & 0.241 & 0.241    & 0.686 & 0.254 & 0.686                   & 0.241 & 0.686 & 0.269    & 0.686                    \\
 	glycineriboswitch,F.nucleatum & 0.568 & 0.627 & 0.868    & 0.952 & 0.868 & 0.658    & 0.868 & 0.868 & 0.306    & 0.313 & 0.395 & 0.313                   & 0.593 & 0.593 & 0.693    & 0.6                      \\
 	cidGMPriboswitch,V.Cholerae   & 0.654 & 0.77  & 0.654    & 0.77  & 0.654 & 0.77     & 0.654 & 0.77  & 0.77     & 0.77  & 0.667 & 0.77                    & 0.77  & 0.72  & 0.667    & 0.77                     \\
 	P4P6domain                    & 0.864 & 0.856 & 0.881    & 0.864 & 0.864 & 0.856    & 0.864 & 0.856 & 0.837    & 0.808 & 0.773 & 0.714                   & 0.845 & 0.808 & 0.781    & 0.722                    \\
 	adenineriboswitch,add         & 1     & 0.956 & 1        & 0.977 & 1     & 0.956    & 1     & 1     & 1        & 0.333 & 0.41  & 1                       & 1     & 0.333 & 0.356    & 0.306                    \\
 	tRNAphenylalanineyeast        & 0.286 & 0.746 & 0.976    & 0.746 & 0.976 & 0.746    & 0.976 & 0.976 & 0.976    & 0.976 & 0.976 & 0.976                   & 0.334 & 0.976 & 0.976    & 0.976                    \\ \midrule
 	Average                       & 0.604 & 0.7   & 0.77     & 0.76  & 0.77  & 0.705    & 0.77  & 0.785 & 0.69     & 0.65  & 0.58  & 0.74                    & 0.63  & 0.69  & 0.62     & 0.68                     \\ \bottomrule
 \end{tabular}
\end{adjustbox}
\caption{GM of the structures predicted with \OurTool{}, compared to predictions of \Software{RNAfold} in energy mimization (MFE) and accuracy maximization (MEA) modes.}
%( ref. $supplementary\_data/benchmark\_6RNAs$)}
\end{table}
\subsection{Stablity analysis}
\begin{table}[H]
\centering
\begin{adjustbox}{max width=1\textwidth}
 \begin{tabular}{lHHHHHHHHHHHHHHHHHHHHH}\toprule
 &\multicolumn{3}{l}{ NMIA} & \multicolumn{3}{l}{ DMS} & \multicolumn{3}{l}{CMCT} &\multicolumn{3}{l}{ NMIA+DMS} & \multicolumn{3}{l}{ NMIA+CMCT} & \multicolumn{3}{l}{DMS+CMCT}& \multicolumn{3}{l}{NMIA+DMS+CMCT} \\
% & \multicolumn{3}{l}{ GM }  & \multicolumn{3}{l}{ GM } & \multicolumn{3}{l}{ GM }& \multicolumn{3}{l}{ GM }  & \multicolumn{3}{l}{ GM } & \multicolumn{3}{l}{ GM } &\multicolumn{3}{l}{ GM }\\

Replicates& \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3} & \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3}& \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3} & \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3}& \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3} & \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3}& \multicolumn{1}{l}{Run 1} & \multicolumn{1}{l}{Run 2} & \multicolumn{1}{l}{Run 3}\\
\midrule
5SRNA,E.Coli	&	0.247	&	0.25	&	0.247	&	0.254	&	0.244	&	0.244	&	0.238	&	0.238	&	0.238	&	0.247	&	0.244	&	0.247	&	0.257	&	0.238	&	0.238	&	0.247	&	0.247	&	0.244	&	0.241	&	0.241	&	0.241	\\
glycineriboswitch,F.nucleatum	&	0.868	&	0.868	&	0.868	&	0.952	&	0.952	&	0.952	&	0.868	&	0.634	&	0.627	&	0.868	&	0.868	&	0.868	&	0.868	&	0.868	&	0.868	&	0.667	&	0.658	&	0.658	&	0.868	&	0.868	&	0.868	\\
cidGMPriboswitch,V.Cholerae	&	0.631	&	0.631	&	0.631	&	0.77	&	0.77	&	0.77	&	0.667	&	0.77	&	0.77	&	0.631	&	0.631	&	0.756	&	0.77	&	0.77	&	0.77	&	0.77	&	0.77	&	0.77	&	0.77	&	0.77	&	0.77	\\
P4P6domain	&	0.864	&	0.881	&	0.881	&	0.864	&	0.864	&	0.856	&	0.856	&	0.856	&	0.856	&	0.864	&	0.864	&	0.864	&	0.864	&	0.864	&	0.864	&	0.856	&	0.856	&	0.856	&	0.856	&	0.856	&	0.856	\\
adenineriboswitch,ad	&	1	&	1	&	1	&	0.956	&	0.956	&	0.956	&	1	&	0.956	&	0.956	&	1	&	1	&	1	&	1	&	1	&	1	&	0.956	&	0.956	&	0.956	&	1	&	1	&	1	\\
tRNAphenylalanineyeast	&	0.953	&	0.953	&	0.976	&	0.976	&	0.746	&	0.976	&	0.746	&	0.746	&	0.746	&	0.976	&	0.976	&	0.976	&	0.976	&	0.976	&	0.97	&	0.746	&	0.746	&	0.746	&	0.746	&	0.976	&	0.976	\\

\bottomrule
\end{tabular}
\end{adjustbox}
\caption{GM of \OurTool predictions over three consecutive runs from up 3 sources of probing data.}
%( ref. $supplementary\_data/benchmark\_6RNAs$)
\end{table}


%\section{IPANEMAP Vs Rsample}
\section{Mono probing benchmark (Hajdin~\emph{et al}~\cite{Hajdin2013} dataset)}

 \begin{table}[H]
{\centering
\begin{adjustbox}{max width=1\textwidth}
 \begin{tabular}{@{}llll@{}}
 	\toprule
 	RNA ID & Description                      \\ \midrule
 	A      & Pre-Q1 riboswitch, B. subtilis      & 	M      &SARS corona virus pseudoknot       \\
 	B      & 5’ domain of 16S rRNA, E. coli            & 	N      & 	5S rRNA, E. coli                      \\
 	C      &	Signal recogniton particle RNA human & 	O      & cyclic-di-GMP riboswitch, V. cholerae   \\
 	D      & 	Group I intron, Azoarcus sp.          & 	P      & 5’ domain of 23S rRNA, E. coli            \\
 	E      & 	HIV 15 prime pseudoknot domain       & 	Q      & RNase PB. subtilis                \\
 	F      & Telomerase pseudoknot human       & 	R      & 	Group I Intron, T. thermophila        \\
 	G      & 	SAM I riboswitch, T. tengcongensis   & 	S      & 	Hepatitis C virus IRES domain       \\
 	H      & 	tRNA(phe), E. coli                & 	T      & 	5' domain of 16S rRNA, H. volcanii      \\
 	I      & P546 domain, bI3 group I intron       & 	U      & 	Group II intron, O. iheyensis         \\
 	J      & TPP riboswitch, E. coli              & 	V      & 	tRNA(asp), yeast                    \\
 	K      &	Adenine riboswitch, V. vulnificus    & 	W      & 	Lysine riboswitch, T. maritima        \\
 	L      & Fluoride riboswitch, P. syringae     & 	X      & M-Box riboswitch, B. subtilis          \\
 	\bottomrule
 \end{tabular}
\end{adjustbox}\\}

\caption{Labels used as shorthands for the 24 RNAs in the Hajdin \emph{ et al.} dataset~\cite{Hajdin2013}.}
\label{clyu}

\end{table}

\subsection{Detailed results of \OurTool{} and \Software{Rsample}}
\begin{table}[H]
	{\centering \begin{adjustbox}{max width=\textwidth,max totalheight=\textheight,keepaspectratio}
			\begin{tabular}{llllHllH}
				\toprule
				&        & \multicolumn{3}{l}{\Software{RSample}}    & \multicolumn{3}{l}{\Software{IPANEMAP}}   \\
				{RNA} & Length & Sensitivity & PPV    & \multicolumn1c{GM} & Sensitivity & PPV    & \multicolumn1c{GM} \\ \midrule
				A     & 34     & 0.625       & 1      & 0.791              & 0.625       & 1      & 0.791              \\
				B     & 530    & 0.6622      & 0.6203 & 0.641              & 0.8986      & 0.8062 & 0.851              \\
				C     & 301    & 0.03        & 0.0313 & 0.031              & 0.6         & 0.6    & 0.6                \\
				D     & 214    & 0.6349      & 0.6667 & 0.651              & 0.7302      & 0.8519 & 0.789              \\
				E     & 500    & 0.6382      & 0.7823 & 0.707              & 0.5263      & 0.5517 & 0.539              \\
				F     & 47     & 0.4         & 0.75   & 0.548              & 0.6         & 0.75   & 0.671              \\
				G     & 118    & 0.7179      & 0.8    & 0.758              & 0.7692      & 0.8333 & 0.801              \\
				H     & 76     & 0.4762      & 0.5556 & 0.514              & 0.7619      & 0.7619 & 0.762              \\
				I     & 155    & 0.6786      & 0.8261 & 0.749              & 0.9821      & 0.9821 & 0.982              \\
				J     & 79     & 0.7727      & 0.85   & 0.81               & 0.9545      & 0.875  & 0.914              \\
				K     & 71     & 1           & 1      & 1                  & 1           & 0.9545 & 0.977              \\
				L     & 66     & 0.5625      & 0.6923 & 0.624              & 0.625       & 0.7143 & 0.668              \\
				M     & 79     & 0.6538      & 0.8095 & 0.727              & 0.6538      & 0.68   & 0.667              \\
				N     & 120    & 0.2857      & 0.2632 & 0.274              & 0.9714      & 0.9189 & 0.945              \\
				O     & 97     & 0.9286      & 0.9286 & 0.929              & 0.9286      & 0.8966 & 0.912              \\
				P     & 511    & 0.9328      & 0.7762 & 0.851              & 0.8908      & 0.7681 & 0.827              \\
				Q     & 401    & 0.5826      & 0.5583 & 0.57               & 0.7304      & 0.75   & 0.74               \\
				R     & 425    & 0.8397      & 0.8271 & 0.833              & 0.9084      & 0.8815 & 0.895              \\
				S     & 336    & 0.5481      & 0.6129 & 0.58               & 0.8077      & 0.866  & 0.836              \\
				T     & 473    & 0.7986      & 0.7143 & 0.755              & 0.8819      & 0.8194 & 0.85               \\
				U     & 412    & 0.5379      & 0.6174 & 0.576              & 0.75        & 0.8462 & 0.797              \\
				V     & 75     & 0.619       & 0.6842 & 0.651              & 0.619       & 0.5652 & 0.591              \\
				W     & 174    & 0.8095      & 0.8644 & 0.836              & 0.8095      & 0.9808 & 0.891              \\
				X     & 154    & 0.875       & 0.913  & 0.894              & 0.875       & 0.913  & 0.894              \\ \bottomrule
			\end{tabular}
		\end{adjustbox}\\}
\caption{\OurTool~ prediction performance compared to \Software{RSample} tool}
\end{table}

\subsection{Stability analysis}

\begin{table}[H]
	\begin{adjustbox}{max width=1\textwidth,max totalheight=\textheight,keepaspectratio}
		\begin{tabular}{crHHHHHHHHHHHrHH}
			\toprule
			\textbf{RNA} & Length &  \multicolumn1c{Run1} &\multicolumn1c{Run2}&\multicolumn1c{Run3}&\multicolumn1c{Run4}&\multicolumn1c{Run5}&\multicolumn1c{Run6}&\multicolumn1c{Run7}&\multicolumn1c{Run8}&\multicolumn1c{Run9}&\multicolumn1c{Run10}& \multicolumn1c{Average}& \multicolumn1c{Std Dev}&\multicolumn1c{\textbf{MFE-RNAfold}} & \multicolumn1c{\textbf{MEA-RNAfold}}\\
			\midrule
			
			A	&	34	&	0.791	&	0.791	&	0.791	&		0.791	&	0.791	&	0.791	&	0.791	&	0.791	&	0.791	&	0.722	&	0.7841	&	0.0207	&	0.791	&	0.791	\\
			B	&	530	&	0.851	&	0.851	&	0.851	&		0.851	&	0.851	&	0.851	&	0.851	&	0.851	&	0.851	&	0.851	&	0.851	&	0	&	0.851	&	0.849	\\
			C	&	301	&	0.6	&	0.6	&	0.6	&		0.6	&	0.6	&	0.6	&	0.6	&	0.6	&	0.6	&	0.6	&	0.6	&	0	&	0.6	&	0.597	\\
			D	&	214	&	0.789	&	0.789	&	0.789	&		0.781	&	0.781	&	0.781	&	0.781	&	0.781	&	0.781	&	0.781	&	0.7834	&	0.0037	&	0.768	&	0.761	\\
			E	&	500	&	0.539	&	0.539	&	0.539	&		0.539	&	0.539	&	0.539	&	0.539	&	0.535	&	0.541	&	0.515	&	0.5364	&	0.0073	&	0.562	&	0.524	\\
			F	&	47	&	0.671	&	0.671	&	0.671	&		0.671	&	0.671	&	0.671	&	0.671	&	0.671	&	0.671	&	0.671	&	0.671	&	0	&	0.671	&	0.671	\\
			G	&	118	&	0.801	&	0.801	&	0.801	&		0.801	&	0.801	&	0.801	&	0.801	&	0.801	&	0.801	&	0.801	&	0.801	&	0	&	0.892	&	0.879	\\
			H	&	76	&	1	&	1	&	1	&		1	&	1	&	1	&	1	&	0.762	&	0.762	&	0.781	&	0.9305	&	0.1063	&	1	&	1	\\
			I	&	155	&	0.982	&	0.982	&	0.982	&		0.982	&	0.982	&	0.982	&	0.982	&	0.982	&	0.982	&	0.964	&	0.9802	&	0.0054	&	0.964	&	0.982	\\
			J	&	79	&	0.914	&	0.914	&	0.914	&		0.914	&	0.914	&	0.914	&	0.914	&	0.914	&	0.597	&	0.889	&	0.8798	&	0.0946	&	0.914	&	0.914	\\
			K	&	71	&	1	&	1	&	1	&		1	&	1	&	1	&	1	&	0.977	&	0.977	&	0.977	&	0.9931	&	0.0105	&	1	&	1	\\
			L	&	66	&	0.668	&	0.668	&	0.668	&		0.668	&	0.668	&	0.668	&	0.668	&	0.668	&	0.668	&	0.668	&	0.668	&	0	&	0.668	&	0.668	\\
			M	&	79	&	0.706	&	0.706	&	0.706	&		0.706	&	0.706	&	0.706	&	0.721	&	0.721	&	0.667	&	0.745	&	0.709	&	0.0184	&	0.692	&	0.692	\\
			N	&	120	&	0.945	&	0.945	&	0.945	&		0.945	&	0.945	&	0.945	&	0.812	&	0.812	&	0.812	&	0.823	&	0.8929	&	0.064	&	0.945	&	0.914	\\
			O	&	97	&	0.912	&	0.912	&	0.912	&		0.912	&	0.912	&	0.912	&	0.912	&	0.912	&	0.929	&	0.929	&	0.9154	&	0.0068	&	0.948	&	0.948	\\
			P	&	511	&	0.824	&	0.824	&	0.824	&		0.821	&	0.821	&	0.818	&	0.818	&	0.833	&	0.83	&	0.827	&	0.824	&	0.0046	&	0.831	&	0.828	\\
			Q	&	401	&	0.74	&	0.74	&	0.74	&		0.74	&	0.74	&	0.74	&	0.74	&	0.74	&	0.74	&	0.74	&	0.74	&	0	&	0.813	&	0.75	\\
			R	&	425	&	0.895	&	0.895	&	0.895	&		0.895	&	0.895	&	0.895	&	0.895	&	0.895	&	0.895	&	0.895	&	0.895	&	0	&	0.885	&	0.92	\\
			S	&	336	&	0.836	&	0.836	&	0.836	&		0.836	&	0.836	&	0.836	&	0.836	&	0.836	&	0.836	&	0.831	&	0.8355	&	0.0015	&	0.831	&	0.831	\\
			T	&	473	&	0.85	&	0.85	&	0.85	&		0.85	&	0.85	&	0.85	&	0.85	&	0.85	&	0.85	&	0.85	&	0.85	&	0	&	0.863	&	0.863	\\
			U	&	412	&	0.929	&	0.929	&	0.929	&		0.929	&	0.81	&	0.81	&	0.81	&	0.806	&	0.806	&	0.925	&	0.8683	&	0.06	&	0.929	&	0.843	\\
			V	&	75	&	0.591	&	0.591	&	0.591	&		0.591	&	0.591	&	0.591	&	0.591	&	0.591	&	0.591	&	0.901	&	0.622	&	0.093	&	0.591	&	0.591	\\
			W	&	174	&	0.891	&	0.891	&	0.891	&		0.891	&	0.891	&	0.823	&	0.823	&	0.823	&	0.823	&	0.823	&	0.857	&	0.034	&	0.842	&	0.823	\\
			X	&	154	&	0.894	&	0.894	&	0.894	&		0.894	&	0.894	&	0.894	&	0.894	&	0.894	&	0.894	&	0.894	&	0.894	&	0	&	0.894	&	0.894	\\
			
			
			\bottomrule
		\end{tabular}
	\end{adjustbox}
	\caption{MCC of predicted structures with \OurTool{} through 10 runs. The Average and the standard deviation (Sd) of the MCC over the 10 runs are reported alongside with the MCC of the MFE and MEA structures obtained with \Software{RNAfold}.}.
\end{table}

\Heatset{min=0.4,  
	max=1,   
	max colour=Aquamarine3, % colour at maximum
	min colour=BadCol,      % colour at minimum
	Min colour=BadCol, % colour for values below min
	Max colour=Aquamarine3   % colour for values above max
}


%  \Heatset{min=0,  
%	max=1,   
%	max colour=Aquamarine3, % colour at maximum
%	min colour=BadCol,      % colour at minimum
%	Min colour=BadCol, % colour for values below min
%	Max colour=Aquamarine3   % colour for values above max
%}




%  \HeatSecset{min=0,   
%	max=.1,   
%	max colour=BadCol, % colour at maximum
%	min colour=Aquamarine3,      % colour at minimum
%	Min colour=Aquamarine3, % colour for values below min
%	Max colour=BadCol   % colour for values above max
%}
\newpage
\section{Case-study: Lariat capping ribozyme of {\itshape Didymium iridis}}
\subsection{Validation of accessibility profiles: Agreement with native and predicted structures}
We assess and validate the overall quality of probing data by defining a natural notion of {\bf agreement} between a reactivity profile and a structure. To that purpose, we consider a {\bf proportion $\alpha$ of the most and least reactive positions}, initially considering $\alpha=30\%$. Then, we investigate whether positions associated with extreme reactivity values are indeed paired or unpaired within the native structure.

\paragraph{Agreement metrics.}More precisely, given a secondary structure $S$ of length $n$, we define the {\bf 1D projection} of $S$ as the vector $v_S$ such that
$$ \forall i\in[1,n], v_S(i)=\begin{cases} 1 &\text{if $i$ unpaired in $S$}\\ 0 &\text{otherwise.} \end{cases} $$
Given a vector $r$ of reactivities, we define a {\bf discrete reactivity vector} $d_{r}^{\alpha}$ as
$$ \forall i\in[1,n], d_{r}^{\alpha}(i)=\begin{cases} 1 &\text{if $r_i$ within the $\alpha$ largest reactivity values},\\ 
0 & \text{if $r_i$ within the $\alpha$ lowest reactivity values},\\
1/2 &\text{otherwise (including if not available).} \end{cases} $$
By focusing on the highest and lowest values, we avoid the influence of outliers, and hope to partially mitigate the natural variability of probing reactivities.

We now define the {\bf normalized divergence} $||r,S||$ of a reactivity profile $r$ from a structure $S$ as
$$ ||r,S|| = \frac{\sum_{i=1}^n |v_S(i)-d_r(i)|-(1-2\alpha)\times \frac{n}{2}}{n\times 2\alpha}.$$
A correcting term counterbalances the $1/2$ associated with medium values, leading to an overall divergence of $(1-2\alpha)\times \frac{n}{2}$ regardless of the quality of the profile. The divisor term of $n\times 2\alpha$ accounts for the fact that, after compensating for the contribution of the $[\alpha,1-\alpha]$ percentile, only a proportion $2\alpha$ of the original values is effectively covered by the sum. Implementing those two corrections leads to a final value scaling between $0$ (all highly reactive pos. unpaired, all poorly reactive pos. paired) and $1$ (all highly reactive pos. paired, all poorly reactive pos. unpaired). 

Finally, we simply define the {\bf agreement of a reactivity profile $r$ with a structure $S$} as
$$ \text{Agreement}(r,S) = 1- ||r,S||.$$

\begin{figure}[t!]
	\Heatset{min=30,  
		max=80,
		min colour=BadCol,      % colour at minimum
		Min colour=BadCol, % colour for values below min
		max colour=Aquamarine3, % colour at maximum
		Max colour=Aquamarine3   % colour for values above max
	}
	\HeatSecset{min=60,  
		max=85,   
		max colour=Aquamarine3, % colour at maximum
		min colour=BadCol,      % colour at minimum
		Min colour=BadCol, % colour for values below min
		Max colour=Aquamarine3   % colour for values above max
	}
		{\centering\resizebox{.5\linewidth}{!}{\begin{tabular}{@{}lllHHS}
			\toprule
			    &          &                    & \multicolumn{2}{{@{}c@{}}}{Agreement$(R,S)$ (\%)}                                  &                               \\
			Num &  Cluster        & Condition $R$      & \multicolumn1{@{}c}{$S:=\mathcal{N}$} & \multicolumn1{@{}c}{$S:=\mathcal{P}$} & \multicolumn1{c@{}}{MCC\%} \\ \toprule
			1   & \Bull{C} & \OneMSevILUMg      & 70                                   & 77                                   & 85                            \\
			2   & \Bull{A} & \OneMSevILU        & 60                                   & 68                                   & 84                            \\
			3   & \Bull{A} & \NMIAMg            & 73                                   & 74                                   & 81                            \\
			4   & \Bull{A} & \NMIA              & 62                                   & 70                                   & 80                            \\
			5   & \Bull{C} & \OneMSevILUThreeMg & 62                                   & 58                                   & 74                            \\
			6   & \Bull{E} & \NMIAMgCE          & 61                                   & 54                                   & 73                            \\
			7   & \Bull{G} & \NAIMg             & 61                                   & 62                                   & 73                            \\
			8   & \Bull{H} & \BzCNMg            & 70                                   & 62                                   & 71                            \\
			9   & \Bull{E} & \OneMSevMgCE       & 58                                   & 52                                   & 70                            \\
			10  & \Bull{G} & \CMCTMg            & 55                                   & 51                                   & 70                            \\
			11  & \Bull{D} & \OneMSevILUThree   & 61                                   & 67                                   & 64                            \\
			12  & \Bull{F} & \DMSMg             & 62                                   & 62                                   & 63                            \\
			13  & \Bull{B} & \NMIACE            & 62                                   & 69                                   & 61                            \\
			14  & \Bull{B} & \OneMSevCE         & 59                                   & 62                                   & 60                            \\
			15  & \Bull{B} & \BzCN              & 54                                   & 49                                   & 60                            \\
			16  & \Bull{B} & \NAICE             & 56                                   & 59                                   & 60                            \\ \bottomrule
		\end{tabular}}\\}
		
%	\end{subfigure}\quad
%	\begin{subfigure}[b]{0.52\textwidth}
%		{\includegraphics[width=\textwidth,clip,trim=0 0 1.8cm 1.5cm]{graphs/didy/Agreement-Distribution.pdf}}
%		
%		\caption{Agreement distribution with random reactivity profiles.\label{fig:distribution_agreement}}
%		\label{fig:mouse}
%	\end{subfigure}
	\caption{Analysis of agreements for reactivity profiles generated in various experimental conditions with both native ($\mathcal{N}$) and \OurTool{}-predicted ($\mathcal{P}$) structure.\label{tab:experiment_agreement}}
\end{figure}



\paragraph{Results.}We computed, and report in Table~\ref{tab:experiment_agreement}, the agreement of our 16 reactivity profiles with both the native and \OurTool{} prediction when considering the $\alpha$ percentile of highest/lowest reactivities ($\alpha=30\%$). A cursory inspection of the table reveals a {\bf good overall agreement of the measured reactivities with the native structure}, involving $61\%$ of the positions on average. This agreement increases to an average $70\%$ when $\alpha$ is set to $15\%$, indicating a good correlation between raw reactivities values and paired/unpaired status in the native structure. Also, expectedly, the agreement to the native structure appears to be positively correlated with the MCC induced by the reactivity profile. However, this relationship is not strict, as revealed by the $60\%$ agreement to the native observed for \OneMSevILU. While superficially surprising, this observation is consistent with the notion that probing data do not need to inform on many different positions to improve the quality of predictions.

We tested the {\bf statistical significance} of our observation by computing  the agreement distribution in the absence of structural information. To that purpose, we generated random reactivity profiles, drawing a random reactivity uniformly in $[0,1]$ for each position. The resulting distribution can be perfectly fitted to a Normal distribution with mean agreement $\mu=49.96\%$ and standard deviation $\sigma=4.51\%$.  {\bf All reactivity profiles showed better agreement with the native structure than a random reactivity profile}, by a margin of \Def{up to five standard deviations}, and always at least one standard deviation, better than the expected value of random profiles. This leads to \Def{significantly enriched agreements} (P-values(Agreement)$<$5\%) for 13 out of our 16 reactivity profiles, the remaining outliers having associated P-values of 18.5\%, 13.2\% and 9\% respectively. Overall, these observations support the notion of informative probing experiments.

	\newcommand{\SR}[4]{{\begin{minipage}{.48\textwidth}\centering 
			\includegraphics[angle=-90,width=\textwidth, trim=0 1.2cm 0 .5cm,clip]{graphs/didy/reactivities/#1}\\
			#2 -- #4 (Cluster #3)\end{minipage}}}

\begin{figure}	
	{\centering \SR{1M7_Ilu_d1_Mg.pdf}{1    }{\Bull{C}  }{\OneMSevILUMg  }
	\SR{1M7_Ilu_d1.pdf}{2    }{\Bull{A}  }{\OneMSevILU    }\\    
	\SR{NMIA_Mg_IT.pdf}{3    }{\Bull{A}  }{\NMIAMg         }   
	\SR{NMIA_IT.pdf}{4    }{\Bull{A}  }{\NMIA            }\\  
	\SR{1M7_Ilu_d3_Mg.pdf}{5    }{\Bull{C}  }{\OneMSevILUThreeMg} 
	\SR{NMIA_Mg_CE.pdf}{6    }{\Bull{E}  }{\NMIAMgCE          }\\
	\SR{NAI_Mg_CE.pdf}{7    }{\Bull{G}  }{\NAIMg             }
	\SR{BzCN_Mg_CE.pdf}{8    }{\Bull{H}  }{\BzCNMg            }\\}
\caption{Projections of reactivities onto native structure (conditions 1-8 of 16).}
\end{figure}
\begin{figure}
	\SR{1M7_Mg_CE.pdf}{9    }{\Bull{E}  }{\OneMSevMgCE       }
	\SR{CMCT.pdf}{10   }{\Bull{G}  }{\CMCTMg            }
	\SR{1M7_Ilu_d3.pdf}{11   }{\Bull{D}  }{\OneMSevILUThree   }
	\SR{DMS.pdf}{12   }{\Bull{F}  }{\DMSMg             }
	\SR{NMIA_CE.pdf}{13   }{\Bull{B}  }{\NMIACE            }
	\SR{1M7_CE.pdf}{14   }{\Bull{B}  }{\OneMSevCE         }
	\SR{BzCN_CE.pdf}{15   }{\Bull{B}  }{\BzCN              }
	\SR{NAI_CE.pdf}{16   }{\Bull{B}  }{\NAICE   }
\caption{Projections of reactivities onto native structure (conditions 9-16 of 16).}
\end{figure}


\newcommand{\MyScale}{.26}

\subsection{Correlation analysis of raw reactivities profiles}
\begin{figure}[H]
	\includegraphics[width=.54\textwidth]{graphs/didy/reactivity_correlation.pdf}\hfill
	\includegraphics[width=.45\textwidth]{graphs/didy/bi_clustering.pdf}
	\caption{Correlation of reactivity profiles (left) and induced (pseudo-)ensemble distance (right). Conditions clustered in our analysis due to their impact of the pseudo-Boltzmann ensemble (black boxes, right) do not necessarily correlate very well in term of raw reactivity (white boxes, left).\label{fig:profile_correlation}}
\end{figure}

We compared raw reactivity data produced across various conditions, in order to test their overall compatibility, investigate their main determinants and compare their impacts on predictions.

\paragraph{Correlation metrics and rationale.}To that purpose, we consider the Pearson product-moment \Def{correlation coefficients} $C_{r,r'}$ of two reactivity profile $r$ and $r'$, defined as
$$ C_{r,r'} = \frac{\sum^n_{i=1}\left(r_i - \mu_r\right)\left(r'_i - \mu_{r'}\right)}{\sqrt{\sum ^n _{i=1}\left(r_i - \mu_{r}\right)^2} \sqrt{\sum ^n _{i=1}\left(r'_i - \mu_{r'}\right)^2}},$$ 
where $\mu_{r}$ and $\mu_{r'}$  denote the mean reactivity within $r$ and $r'$ respectively. 

The choice of the \Def{correlation coefficient as a measurement of pairwise similarity} for probing profiles is motivated by differences in the distributions of reactivities across experimental conditions and protocols. 
Indeed, $C_{r,r'}$ remains unchanged by the application of any rescaling, through an affine transformation with positive slope, to either $r$, $r'$ (or both). Moreover, this metrics does not scale with the number of positions.
Thus, we can restrict the sums and mean values to positions for which reactivities are available in both experiments, and still compare the coefficient with another pair of reactivity profiles. This property is important in order to tolerate \Def{missing values} introduced by experimental artifacts and reagent properties, while still being able to compare the coefficients.

\paragraph{Results.} We computed the correlation coefficient for all pairs of reactivity profiles, and report the results in the left matrix of Figure~\ref{fig:profile_correlation}. For the sake of comparison, we also remind in this Figure the Ensemble Distance matrix, indicating the difference in impact on the computational predictions of each pair of conditions.

We first observe that almost all of the pairs of reactivity profiles show a positive correlation, as expected from their observation of a common structure. A notable exception is \OneMSevMgCE, having average negative correlation with all its peers (except for \NMIAMgCE). Its outlier status is confirmed by its large ensemble distance to other conditions, yet does not prevent it to achieve a respectable MCC of 70\%. Finally, the perfectly negative correlation between \CMCTMg{} and \DMSMg{} can be dismissed as a computational artifact, since CMCT and DMS react with different nucleotides, and the correlation cannot be computed.

Moreover, we observe that conditions clustered from the Ensemble Distance metrics (due to their impact of the pseudo-Boltzmann ensemble) do not always correlate in term of raw reactivity, with very poor correlations occurring within clustered conditions. This essentially reflects the fact that similar profiles may induce very similar pseudo-Boltzmann ensembles, \emph{ie} different premises sometimes imply the same conclusions. 

Conversely, one observes very few strong correlations outside of clusters. In fact, the only two exceptions are \OneMSevILU and \OneMSevILUThree which, despite belonging to different clusters and achieving different qualities of prediction, correlate almost perfectly with each other, and with other reactivity profiles produced using 1M7/SHAPEMap. Other natural candidates for explaining the reactivity correlations (Map vs CE, +Mg vs -Mg\ldots) do not appear to be supported by this analysis.

Overall, this analysis confirms the consistency of the produced probing profiles (positive correlation), but also suggests a very delicate effect of reactivity data on the downstream predictions, with very different profiles leading to similar predictions (\OneMSevILU), and similar profiles inducing very different ones (\OneMSevILU vs \OneMSevILUThree).


\subsection{Dominant conformations within main clusters}

\begin{figure}[H]
{\centering
  \sf	\begin{adjustbox}{max width=\textwidth,max height=\textheight,keepaspectratio}
\begin{tabular}{@{}ccc@{}}

%\toprule
 \includegraphics[scale=\MyScale]{graphs/Supp_structures/native_structure}&
 \includegraphics[scale=\MyScale]{graphs/Supp_structures/1M7ILU_NMIAMg}& 
 \includegraphics[scale=\MyScale]{graphs/Supp_structures/1M7ILUMg_1M7ILU3Mg}\\
Native&\OneMSevILU&\OneMSevILUMg\\ 
structure &+ \NMIAMg&+ \OneMSevILUThreeMg\\[2em]
 \includegraphics[scale=\MyScale]{graphs/Supp_structures/1M7MgCE_NMIAMgCE}
 &
 \includegraphics[scale=\MyScale]{graphs/Supp_structures/1M7CE_NMIACE}
 & \includegraphics[scale=\MyScale]{graphs/Supp_structures/NaiMg_CMCTMg}\\
\OneMSevMgCE& \OneMSevCE &\NAIMg\\
+ \NMIAMgCE&+ \NMIACE &+ \CMCTMg\\
\end{tabular}
\end{adjustbox}\\}

\caption{Predicted structures with conditions belonging to the same cluster}

\end{figure}

\subsection{Prediction accuracy for pairs of conditions}


\begin{table}[H]
\newcommand{\MA}[1]{\multicolumn{1}{c}{\Acc{#1}}}
\newcommand{\Acc}[1]{{\bfseries #1}}
\begin{adjustbox}{max width=\textwidth,max totalheight=\textheight,keepaspectratio}
\begin{tabular}{lHHHHHHHHHHHHHHHHH}
	\toprule
	                              & \multicolumn{1}{l}{Mono} & \MA{1} & \MA{2} & \MA{3} & \MA{4} & \MA{5} & \MA{6} & \MA{7} & \MA{8} & \MA{9} & \MA{10} & \MA{11} & \MA{12} & \MA{13} & \MA{14} & \MA{15} & \MA{16} \\ \midrule
	%Mono                         &                          & 0.85   & 0.84   & 0.81   & 0.8    & 0.74   & 0.73   & 0.73   & 0.71   & 0.7    & 0.7     & 0.64    & 0.63    & 0.61    & 0.6     & 0.6     & 0.6     \\
	\Acc{1} -- \OneMSevILUMg      & 0.85                     & 0.85   & 0.84   & 0.8    & 0.82   & 0.82   & 0.82   & 0.75   & 0.79   & 0.78   & 0.79    & 0.79    & 0.85    & 0.61    & 0.79    & 0.79    & 0.85    \\
	\Acc{2} -- \OneMSevILU        & 0.84                     &        & 0.84   & 0.82   & 0.82   & 0.7    & 0.83   & 0.72   & 0.82   & 0.85   & 0.83    & 0.84    & 0.84    & 0.84    & 0.6     & 0.84    & 0.6     \\
	\Acc{3} -- \NMIAMg            & 0.81                     &        &        & 0.81   & 0.8    & 0.74   & 0.82   & 0.82   & 0.81   & 0.81   & 0.79    & 0.8     & 0.8     & 0.61    & 0.81    & 0.81    & 0.6     \\
	\Acc{4} -- \NMIA              & 0.8                      &        &        &        & 0.8    & 0.74   & 0.73   & 0.81   & 0.78   & 0.7    & 0.79    & 0.63    & 0.72    & 0.61    & 0.7     & 0.7     & 0.59    \\
	\Acc{5} -- \OneMSevILUThreeMg & 0.74                     &        &        &        &        & 0.74   & 0.74   & 0.7    & 0.74   & 0.74   & 0.7     & 0.63    & 0.74    & 0.74    & 0.74    & 0.74    & 0.74    \\
	\Acc{6} -- \NMIAMgCE          & 0.73                     &        &        &        &        &        & 0.73   & 0.73   & 0.73   & 0.73   & 0.73    & 0.61    & 0.73    & 0.61    & 0.73    & 0.73    & 0.6     \\
	\Acc{7} -- \NAIMg             & 0.73                     &        &        &        &        &        &        & 0.73   & 0.72   & 0.7    & 0.73    & 0.64    & 0.68    & 0.61    & 0.73    & 0.73    & 0.59    \\
	\Acc{8} -- \BzCNMg            & 0.71                     &        &        &        &        &        &        &        & 0.71   & 0.65   & 0.73    & 0.63    & 0.72    & 0.6     & 0.61    & 0.6     & 0.59    \\
	\Acc{9} -- \OneMSevMgCE       & 0.7                      &        &        &        &        &        &        &        &        & 0.7    & 0.7     & 0.7     & 0.7     & 0.61    & 0.61    & 0.6     & 0.6     \\
	\Acc{10} -- \CMCTMg           & 0.7                      &        &        &        &        &        &        &        &        &        & 0.7     & 0.63    & 0.74    & 0.61    & 0.6     & 0.6     & 0.6     \\
	\Acc{11} -- \OneMSevILUThree  & 0.64                     &        &        &        &        &        &        &        &        &        &         & 0.64    & 0.62    & 0.62    & 0.65    & 0.65    & 0.59    \\
	\Acc{12} -- \DMSMg            & 0.63                     &        &        &        &        &        &        &        &        &        &         &         & 0.63    & 0.61    & 0.6     & 0.6     & 0.6     \\
	\Acc{13} -- \NMIACE           & 0.61                     &        &        &        &        &        &        &        &        &        &         &         &         & 0.61    & 0.62    & 0.6     & 0.6     \\
	\Acc{14} -- \OneMSevCE        & 0.6                      &        &        &        &        &        &        &        &        &        &         &         &         &         & 0.6     & 0.61    & 0.59    \\
	\Acc{15} -- \BzCN             & 0.6                      &        &        &        &        &        &        &        &        &        &         &         &         &         &         & 0.6     & 0.6     \\
	\Acc{16} -- \NAICE            & 0.6                      &        &        &        &        &        &        &        &        &        &         &         &         &         &         &         & 0.6     \\ \bottomrule
\end{tabular}
\end{adjustbox}
\caption{MCC of structures prediction with \OurTool{} from any pair of conditions}.
\end{table}


%\begin{table}
%\begin{center}
%\begin{tabular}{ll}
%Sample size & MCC\\
%\toprule
%
%100 & 0.55 \\
%500 & 0.55  \\
%1000 &  0.81\\
%2000 & 	0.55  \\
%3000 &  0.77 \\
%5000 & 	0.55  \\ 
%10000& 0.8 \\
%\end{tabular}
%\end {center}
%\caption{ MCC of predicted structures in absence of probig data when varyingthe sample size }
%\end{table}

\subsection{Histograms of prediction accuracy informed by single-point mutants reactivities (Mutate-and-Map protocol)}
\begin{figure}[H]
%\includegraphics[width=\linewidth]{graphs/histog}
\includegraphics[width=\linewidth]{graphs/histog1}
\caption{Distribution of MCC values over 100 runs from Mutate-and-Map data}
\end{figure}


%\bibliographystyle{bioinformatics}
\bibliographystyle{nar}
%\bibliographystyle{natbib}%compa
\bibliography{biblio}

\end{document}
